diff --git a/src/plugins/contrib/IncrementalSearch/update.bat b/src/plugins/contrib/IncrementalSearch/update.bat
index 353acad..8a18de5 100644
Binary files a/src/plugins/contrib/IncrementalSearch/update.bat and b/src/plugins/contrib/IncrementalSearch/update.bat differ
diff --git a/src/plugins/contrib/ToolsPlus/update.bat b/src/plugins/contrib/ToolsPlus/update.bat
index 25f4898..1016e3f 100644
Binary files a/src/plugins/contrib/ToolsPlus/update.bat and b/src/plugins/contrib/ToolsPlus/update.bat differ
diff --git a/src/update b/src/update
index 31061b7..c6a0cf4 100755
--- a/src/update
+++ b/src/update
@@ -131,10 +131,6 @@ cp -f devel/share/codeblocks/images/settings/*.png output/share/codeblocks/image
 
 rsync -au --exclude-from=excludes.txt plugins/scriptedwizard/resources/ ${RESDIR}/templates/wizard > /dev/null
 rsync -au --exclude-from=excludes.txt plugins/scriptedwizard/resources/ output/share/codeblocks/templates/wizard > /dev/null
-rsync -au --exclude-from=excludes.txt templates/common/ output/share/codeblocks/templates > /dev/null
-rsync -au --exclude-from=excludes.txt templates/unix/ output/share/codeblocks/templates > /dev/null
-rsync -au --exclude-from=excludes.txt templates/common/ ${RESDIR}/templates > /dev/null
-rsync -au --exclude-from=excludes.txt templates/unix/ ${RESDIR}/templates > /dev/null
 cp -f scripts/*.script ${RESDIR}/scripts > /dev/null
 cp -f scripts/*.script output/share/codeblocks/scripts > /dev/null
 cp -f scripts/*.gdb ${RESDIR}/scripts > /dev/null
diff --git a/src/update.bat b/src/update.bat
index f9d215b..68af514 100644
Binary files a/src/update.bat and b/src/update.bat differ
diff --git a/src/update30 b/src/update30
index a26ec93..9304800 100755
--- a/src/update30
+++ b/src/update30
@@ -93,10 +93,6 @@ cp -f plugins/compilergcc/resources/compilers/*.xml output30/share/codeblocks/co
 
 rsync -au --exclude-from=excludes.txt plugins/scriptedwizard/resources/ ${RESDIR}/templates/wizard > /dev/null
 rsync -au --exclude-from=excludes.txt plugins/scriptedwizard/resources/ output30/share/codeblocks/templates/wizard > /dev/null
-rsync -au --exclude-from=excludes.txt templates/common/ output30/share/codeblocks/templates > /dev/null
-rsync -au --exclude-from=excludes.txt templates/unix/ output30/share/codeblocks/templates > /dev/null
-rsync -au --exclude-from=excludes.txt templates/common/ ${RESDIR}/templates > /dev/null
-rsync -au --exclude-from=excludes.txt templates/unix/ ${RESDIR}/templates > /dev/null
 cp -f scripts/*.script ${RESDIR}/scripts > /dev/null
 cp -f scripts/*.script output30/share/codeblocks/scripts > /dev/null
 cp -f scripts/*.gdb ${RESDIR}/scripts > /dev/null
diff --git a/src/update30.bat b/src/update30.bat
index 3c63c75..531cad3 100644
Binary files a/src/update30.bat and b/src/update30.bat differ
diff --git a/src/update30_64.bat b/src/update30_64.bat
index 84380bd..0cebaa2 100644
Binary files a/src/update30_64.bat and b/src/update30_64.bat differ
