diff --git src/plugins/contrib/ToolsPlus/CmdConfigDialog.cpp src/plugins/contrib/ToolsPlus/CmdConfigDialog.cpp
index 9ccc71b..841e7c2 100644
--- src/plugins/contrib/ToolsPlus/CmdConfigDialog.cpp
+++ src/plugins/contrib/ToolsPlus/CmdConfigDialog.cpp
@@ -2,18 +2,22 @@
 // C++ code generated with wxFormBuilder (version Feb  2 2007)
 // http://www.wxformbuilder.org/
 //
-// PLEASE DO "NOT" EDIT THIS FILE!
 ///////////////////////////////////////////////////////////////////////////
 
-#include "wx/wxprec.h"
+// sdk.h must be the first include in the file to work as a PCH
+#include <sdk.h>
 
-#ifdef __BORLANDC__
-#pragma hdrstop
-#endif //__BORLANDC__
+#ifndef CB_PRECOMP
+    // wxWidgets non GUI headers in sdk_common.h
+    // wxWidgets GUI headers in sdk_common.h
+    #include <wx/dialog.h>
+    #include <wx/filedlg.h>
+    #include <wx/notebook.h>
+
+    // CB SDK Headers in sdk_common.h
+    #include <configmanager.h>
+#endif // #ifndef CB_PRECOMP
 
-#ifndef WX_PRECOMP
-#include <wx/wx.h>
-#endif //WX_PRECOMP
 
 #include "ToolsPlus.h"
 #include "CmdConfigDialog.h"
diff --git src/plugins/contrib/ToolsPlus/CmdConfigDialog.h src/plugins/contrib/ToolsPlus/CmdConfigDialog.h
index e049d67..b45113b 100644
--- src/plugins/contrib/ToolsPlus/CmdConfigDialog.h
+++ src/plugins/contrib/ToolsPlus/CmdConfigDialog.h
@@ -2,26 +2,23 @@
 // C++ code generated with wxFormBuilder (version Feb  2 2007)
 // http://www.wxformbuilder.org/
 //
-// PLEASE DO "NOT" EDIT THIS FILE!
 ///////////////////////////////////////////////////////////////////////////
 
 #ifndef __CMDCONFIGDIALOG__
 #define __CMDCONFIGDIALOG__
 
-// Define WX_GCH in order to support precompiled headers with GCC compiler.
-// You have to create the header "wx_pch.h" and include all files needed
-// for compile your gui inside it.
-// Then, compile it and place the file "wx_pch.h.gch" into the same
-// directory that "wx_pch.h".
-#ifdef WX_GCH
-#include <wx_pch.h>
-#else
-#include <wx/wx.h>
-#endif
 
+#include <wx/stattext.h>
+#include <wx/listbox.h>
+#include <wx/sizer.h>
 #include <wx/button.h>
+#include <wx/textctrl.h>
 #include <wx/spinctrl.h>
 #include <wx/choice.h>
+#include <wx/panel.h>
+
+#include <wx/checkbox.h> // wxCheckBox
+
 
 ///////////////////////////////////////////////////////////////////////////
 
@@ -49,7 +46,6 @@
 /// Class CmdConfigDialog
 ///////////////////////////////////////////////////////////////////////////////
 
-#include <sdk.h>
 #include <configurationpanel.h>
 
 class ToolsPlus;
diff --git src/plugins/contrib/ToolsPlus/PipedProcessCtrl.cpp src/plugins/contrib/ToolsPlus/PipedProcessCtrl.cpp
index e72c8af..1b742a4 100644
--- src/plugins/contrib/ToolsPlus/PipedProcessCtrl.cpp
+++ src/plugins/contrib/ToolsPlus/PipedProcessCtrl.cpp
@@ -1,9 +1,23 @@
-#include <wx/notebook.h>
-#include <wx/textctrl.h>
-#include <wx/regex.h>
+// sdk.h must be the first include in the file to work as a PCH
+#include <sdk.h>
+
+#ifndef CB_PRECOMP
+    // wxWidgets non GUI headers in sdk_common.h
+    #include <wx/process.h>
+    #include <wx/regex.h>
+
+    // wxWidgets GUI headers in sdk_common.h
+    #include <wx/notebook.h>
+    #include <wx/textctrl.h>
+
+    // CB SDK Headers in sdk_common.h
+    #include <globals.h>
+    #include <cbeditor.h>
+    #include <configmanager.h>
+    #include <editormanager.h>
+#endif // #ifndef CB_PRECOMP
+
 #include "PipedProcessCtrl.h"
-#include <globals.h>
-#include <cbeditor.h>
 
 ////////////////////////////////////// PipedProcessCtrl /////////////////////////////////////////////
 #define PP_ERROR_STYLE 1
diff --git src/plugins/contrib/ToolsPlus/PipedProcessCtrl.h src/plugins/contrib/ToolsPlus/PipedProcessCtrl.h
index b4c68d2..cea2815 100644
--- src/plugins/contrib/ToolsPlus/PipedProcessCtrl.h
+++ src/plugins/contrib/ToolsPlus/PipedProcessCtrl.h
@@ -1,19 +1,11 @@
 #ifndef PPCTRL_H
 #define PPCTRL_H
 
-#include <wx/wxprec.h>
-
-#ifndef WX_PRECOMP
-	#include <wx/wx.h>
-#endif
-
 #include <wx/process.h>
 #include <wx/aui/aui.h>
 
-#include <sdk.h>
-#ifndef CB_PRECOMP
-    #include <wx/wxscintilla.h>
-#endif
+#include <wx/wxscintilla.h>
+
 #include "ShellCtrlBase.h"
 
 class PipedProcessCtrl;
diff --git src/plugins/contrib/ToolsPlus/ShellCtrlBase.cpp src/plugins/contrib/ToolsPlus/ShellCtrlBase.cpp
index dafd77b..1425774 100644
--- src/plugins/contrib/ToolsPlus/ShellCtrlBase.cpp
+++ src/plugins/contrib/ToolsPlus/ShellCtrlBase.cpp
@@ -1,8 +1,24 @@
-#include <wx/notebook.h>
-#include <wx/textctrl.h>
-#include <wx/regex.h>
+// sdk.h must be the first include in the file to work as a PCH
+#include <sdk.h>
+
+#ifndef CB_PRECOMP
+    // wxWidgets non GUI headers in sdk_common.h
+    #include <wx/regex.h>
+
+    // wxWidgets GUI headers in sdk_common.h
+    #include <wx/menu.h>
+    #include <wx/notebook.h>
+    #include <wx/textctrl.h>
+
+    // CB SDK Headers in sdk_common.h
+    #include <globals.h>
+    #include <manager.h>
+    #include <logmanager.h>
+#endif // #ifndef CB_PRECOMP
+
+#include <wx/aui/aui.h>
+
 #include "ShellCtrlBase.h"
-#include <globals.h>
 
 // The global instance of the shell registry
 ShellRegistry& GlobalShellRegistry()
diff --git src/plugins/contrib/ToolsPlus/ShellCtrlBase.h src/plugins/contrib/ToolsPlus/ShellCtrlBase.h
index c41dfed..8b7dc7c 100644
--- src/plugins/contrib/ToolsPlus/ShellCtrlBase.h
+++ src/plugins/contrib/ToolsPlus/ShellCtrlBase.h
@@ -3,18 +3,12 @@
 
 #include <map>
 
+#include <wx/panel.h>
+#include <wx/timer.h> // wxTimer & wxTimerEvent
 
-#include <wx/wxprec.h>
-
-#ifndef WX_PRECOMP
-    #include <wx/wx.h>
-#endif
-
-
-#include <wx/process.h>
-#include <wx/aui/aui.h>
-#include <sdk.h>
-#include "se_globals.h"
+class wxAuiNotebook;
+class wxAuiNotebookEvent;
+class wxWindow;
 
 // CLASSES DEFINED IN THIS LIBRARY
 class ShellManager; //Manages the collection of Shell Control Widgets allowing user interaction with external processes within a tabbed notepage - usually the main app (or some plugin) will declare a global instance of this manager. See the full declaration below for more detail
diff --git src/plugins/contrib/ToolsPlus/ToolsPlus.cbp src/plugins/contrib/ToolsPlus/ToolsPlus.cbp
index 8a7c33a..92e36a8 100644
--- src/plugins/contrib/ToolsPlus/ToolsPlus.cbp
+++ src/plugins/contrib/ToolsPlus/ToolsPlus.cbp
@@ -6,23 +6,45 @@
 		<Option compiler="gcc" />
 		<Build>
 			<Target title="default">
-				<Option output="..\..\..\devel\share\codeblocks\plugins\ToolsPlus" prefix_auto="1" extension_auto="1" />
-				<Option object_output="..\..\..\.objs\plugins\contrib\ToolsPlus" />
+				<Option output="../../../devel/share/codeblocks/plugins/ToolsPlus" imp_lib="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME).a" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME).def" prefix_auto="1" extension_auto="1" />
+				<Option object_output="../../../.objs/plugins/contrib/ToolsPlus" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --multiple-instance -na -ns -nd -p debug" />
-				<Option host_application="..\..\..\devel\codeblocks.exe" />
+				<Option host_application="../../../devel/codeblocks.exe" />
 				<Option run_host_application_in_terminal="0" />
+				<Option projectIncludeDirsRelation="2" />
 				<Compiler>
 					<Add option="$(#CB_RELEASE_TYPE)" />
+					<Add option="-Winvalid-pch" />
+					<Add option="-DCB_PRECOMP" />
+					<Add directory="../../../.objs/include" />
 				</Compiler>
 				<Linker>
 					<Add library="codeblocks" />
 					<Add library="wxscintilla_cb" />
 					<Add library="wxmsw28$(WX_SUFFIX)" />
 					<Add library="gdi32" />
-					<Add directory="..\..\..\devel" />
-					<Add directory="$(#wx.lib)\gcc_dll$(WX_CFG)" />
+					<Add directory="../../../devel" />
+					<Add directory="$(#wx.lib)/gcc_dll$(WX_CFG)" />
+				</Linker>
+			</Target>
+			<Target title="no_pch">
+				<Option output="../../../devel/share/codeblocks/plugins/ToolsPlus" imp_lib="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME).a" def_file="$(TARGET_OUTPUT_DIR)$(TARGET_OUTPUT_BASENAME).def" prefix_auto="1" extension_auto="1" />
+				<Option object_output="../../../.objs/plugins/contrib/ToolsPlus" />
+				<Option type="3" />
+				<Option compiler="gcc" />
+				<Compiler>
+					<Add option="$(#CB_RELEASE_TYPE)" />
+					<Add option="-DNOPCH" />
+				</Compiler>
+				<Linker>
+					<Add library="codeblocks" />
+					<Add library="wxscintilla_cb" />
+					<Add library="wxmsw28$(WX_SUFFIX)" />
+					<Add library="gdi32" />
+					<Add directory="../../../devel" />
+					<Add directory="$(#wx.lib)/gcc_dll$(WX_CFG)" />
 				</Linker>
 			</Target>
 			<Environment>
@@ -39,25 +61,22 @@
 			<Add option="-mthreads" />
 			<Add option="-fmessage-length=0" />
 			<Add option="-fexceptions" />
-			<Add option="-Winvalid-pch" />
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DHAVE_W32API_H" />
 			<Add option="-D__WXMSW__" />
 			<Add option="-DcbDEBUG" />
-			<Add option="-DCB_PRECOMP" />
-			<Add option="-DWX_PRECOMP" />
 			<Add option="-DwxUSE_UNICODE" />
 			<Add option="-DBUILDING_PLUGIN" />
-			<Add directory="..\..\..\include" />
-			<Add directory="..\..\..\sdk\wxscintilla\include" />
-			<Add directory="..\..\..\include\mozilla_chardet" />
-			<Add directory="..\..\..\include\mozilla_chardet\mfbt" />
-			<Add directory="..\..\..\include\mozilla_chardet\nsprpub\pr\include" />
-			<Add directory="..\..\..\include\mozilla_chardet\xpcom" />
-			<Add directory="..\..\..\include\mozilla_chardet\xpcom\base" />
-			<Add directory="..\..\..\include\mozilla_chardet\xpcom\glue" />
+			<Add directory="../../../include" />
+			<Add directory="../../../sdk/wxscintilla/include" />
+			<Add directory="../../../include/mozilla_chardet" />
+			<Add directory="../../../include/mozilla_chardet/mfbt" />
+			<Add directory="../../../include/mozilla_chardet/nsprpub/pr/include" />
+			<Add directory="../../../include/mozilla_chardet/xpcom" />
+			<Add directory="../../../include/mozilla_chardet/xpcom/base" />
+			<Add directory="../../../include/mozilla_chardet/xpcom/glue" />
 			<Add directory="$(#wx.include)" />
-			<Add directory="$(#wx.lib)\gcc_dll$(WX_CFG)\msw$(WX_SUFFIX)" />
+			<Add directory="$(#wx.lib)/gcc_dll$(WX_CFG)/msw$(WX_SUFFIX)" />
 		</Compiler>
 		<Linker>
 			<Add option="-mthreads" />
@@ -71,7 +90,7 @@
 		<Unit filename="CmdConfigDialog.h" />
 		<Unit filename="PipedProcessCtrl.cpp" />
 		<Unit filename="PipedProcessCtrl.h" />
-		<Unit filename="Resources\dialogs.fbp" />
+		<Unit filename="Resources/dialogs.fbp" />
 		<Unit filename="ShellCtrlBase.cpp" />
 		<Unit filename="ShellCtrlBase.h" />
 		<Unit filename="ToolsPlus.cpp" />
@@ -81,7 +100,6 @@
 		<Unit filename="se_globals.h" />
 		<Unit filename="shellproperties.cpp" />
 		<Unit filename="shellproperties.h" />
-		<Unit filename="update" />
 		<Unit filename="update.bat" />
 		<Extensions>
 			<code_completion />
diff --git src/plugins/contrib/ToolsPlus/ToolsPlus.cpp src/plugins/contrib/ToolsPlus/ToolsPlus.cpp
index b1c8ceb..c275947 100644
--- src/plugins/contrib/ToolsPlus/ToolsPlus.cpp
+++ src/plugins/contrib/ToolsPlus/ToolsPlus.cpp
@@ -1,3 +1,29 @@
+// sdk.h must be the first include in the file to work as a PCH
+#include <sdk.h>
+
+#if !defined(WX_PRECOMP)
+    #include <wx/defs.h>   // wx/dirdlg.h is sometimes missing the normal include of wx/defs.h
+    
+    // wxWidgets GUI headers in wx/wx.h; but, not in sdk_common.h
+    #include <wx/dirdlg.h>
+#endif // #if !defined(WX_PRECOMP)
+
+#ifndef CB_PRECOMP
+    // wxWidgets non GUI headers in sdk_common.h
+
+    // wxWidgets GUI headers in sdk_common.h
+    #include <wx/filedlg.h>
+    #include <wx/menu.h>
+    #include <wx/textdlg.h>
+
+    // CB SDK Headers in sdk_common.h
+    #include <cbproject.h>
+    #include <configmanager.h>
+    #include <editorbase.h>
+    #include <editormanager.h>
+    #include <macrosmanager.h>
+#endif // #ifndef CB_PRECOMP
+
 #include "ToolsPlus.h"
 #include "se_globals.h"
 
diff --git src/plugins/contrib/ToolsPlus/ToolsPlus.h src/plugins/contrib/ToolsPlus/ToolsPlus.h
index 346f928..6cf91c3 100644
--- src/plugins/contrib/ToolsPlus/ToolsPlus.h
+++ src/plugins/contrib/ToolsPlus/ToolsPlus.h
@@ -10,19 +10,14 @@
 #ifndef TOOLSPLUS_H_INCLUDED
 #define TOOLSPLUS_H_INCLUDED
 
-// For compilers that support precompilation, includes <wx/wx.h>
-#include <wx/wxprec.h>
-
-#ifndef WX_PRECOMP
-	#include <wx/wx.h>
-#endif
-
-#include <sdk.h>
 #include <cbplugin.h> // for "class cbPlugin"
 #include "shellproperties.h"
 #include "CmdConfigDialog.h"
 #include "ShellCtrlBase.h"
 
+class wxOutputStream;
+class wxInputStream;
+
 class ToolsPlus : public cbPlugin
 {
     public:
diff --git src/plugins/contrib/ToolsPlus/se_globals.cpp src/plugins/contrib/ToolsPlus/se_globals.cpp
index 52519df..a97f78a 100644
--- src/plugins/contrib/ToolsPlus/se_globals.cpp
+++ src/plugins/contrib/ToolsPlus/se_globals.cpp
@@ -1,3 +1,17 @@
+// sdk.h must be the first include in the file to work as a PCH
+#include <sdk.h>
+
+#ifndef CB_PRECOMP
+    // wxWidgets non GUI headers in sdk_common.h
+    #include <wx/filename.h>
+
+    // wxWidgets GUI headers in sdk_common.h
+
+    // CB SDK Headers in sdk_common.h
+    #include <editorbase.h>
+    #include <editormanager.h>
+#endif // #ifndef CB_PRECOMP
+
 #include "se_globals.h"
 
 wxString GetParentDir(const wxString &path)
diff --git src/plugins/contrib/ToolsPlus/se_globals.h src/plugins/contrib/ToolsPlus/se_globals.h
index 0ea394d..7a26c8c 100644
--- src/plugins/contrib/ToolsPlus/se_globals.h
+++ src/plugins/contrib/ToolsPlus/se_globals.h
@@ -1,24 +1,12 @@
 #ifndef IL_GLOBALS_H
 #define IL_GLOBALS_H
+ 
+#include <logmanager.h>
+#include <globals.h>    // cbMessageBox
 
-#include <wx/wxprec.h>
-
-#ifndef WX_PRECOMP
-	#include <wx/wx.h>
-#endif
+class wxFileName;
+class wxString;
 
-#include <sdk.h>
-#ifndef CB_PRECOMP
-    #include <wx/filename.h>
- 
-    #include <editorbase.h>
-    #include <editormanager.h>
-    #include <logmanager.h>
-    #include <manager.h>
-    #include <macrosmanager.h>
-    #include <configmanager.h>
-    #include <cbproject.h>
-#endif
 
 wxString GetParentDir(const wxString &path);
 
diff --git src/plugins/contrib/ToolsPlus/shellproperties.cpp src/plugins/contrib/ToolsPlus/shellproperties.cpp
index 08f7ade..3b30973 100644
--- src/plugins/contrib/ToolsPlus/shellproperties.cpp
+++ src/plugins/contrib/ToolsPlus/shellproperties.cpp
@@ -1,7 +1,7 @@
-#include "shellproperties.h"
-
 #include <configmanager.h>
 
+#include "shellproperties.h"
+
 #include <wx/arrimpl.cpp>
 WX_DEFINE_OBJARRAY(ShellCommandVec);
 
