diff --git src/plugins/contrib/DoxyBlocks/DoxyBlocks.cbp src/plugins/contrib/DoxyBlocks/DoxyBlocks.cbp
index a16dca5..7e8387c 100644
--- src/plugins/contrib/DoxyBlocks/DoxyBlocks.cbp
+++ src/plugins/contrib/DoxyBlocks/DoxyBlocks.cbp
@@ -18,6 +18,7 @@
 				<Option run_host_application_in_terminal="0" />
 				<Compiler>
 					<Add option="$(#CB_RELEASE_TYPE)" />
+					<Add option="-Winvalid-pch" />
 					<Add option="-pipe" />
 					<Add option="-mthreads" />
 					<Add option="-fmessage-length=0" />
