diff --git src/templates/common/Makefile.am src/templates/common/Makefile.am
index e68f5ad..4af3021 100644
--- src/templates/common/Makefile.am
+++ src/templates/common/Makefile.am
@@ -37,13 +37,4 @@ dist_pkgdata_DATA = console.cbp \
 		sharedlib.template \
 		staticlib.cbp \
 		staticlib-sample.c \
-		staticlib.template \
-		wx-app-ash.cpp \
-		wx-app-ash.h \
-		wx-main-ash.cpp \
-		wx-main-ash.h \
-		wx-main-s.cpp \
-		wx-main-sh.cpp \
-		wx-main-sh.h \
-		wx_pch.h\
-		wxwidgets.png
+		staticlib.template
diff --git src/templates/unix/Makefile.am src/templates/unix/Makefile.am
index 707442e..2ed2f7d 100644
--- src/templates/unix/Makefile.am
+++ src/templates/unix/Makefile.am
@@ -5,6 +5,4 @@ dist_pkgdata_DATA = glfw-static.cbp \
 		irr_linux.cbp \
 		ogre.template \
 		ogre_linux.cbp \
-		sdlapp.cbp \
-		wxwidgets.cbp \
-		wxwidgets.template
+		sdlapp.cbp
diff --git src/templates/win32/Makefile.am src/templates/win32/Makefile.am
index a1c4dbb..8696ea6 100644
--- src/templates/win32/Makefile.am
+++ src/templates/win32/Makefile.am
@@ -33,9 +33,4 @@ dist_pkgdata_DATA = dll.cbp \
 		SmartWin-main.cpp \
 		win32.cbp \
 		win32gui.template \
-		win32-main.cpp \
-		wxwidgets.cbp \
-		wxwidgets_static.cbp \
-		wxwidgets_u.cbp \
-		wxwidgets_static_u.cbp \
-		wxwidgets.template
+		win32-main.cpp
diff --git src/update.bat src/update.bat
index f9d215b..11c2471 100644
--- src/update.bat
+++ src/update.bat
@@ -91,6 +91,21 @@ echo \.svn\      >> excludes.txt
 echo *.gdb       >> excludes.txt
 xcopy /D /y /s plugins\scriptedwizard\resources\* %CB_DEVEL_RESDIR%\templates\wizard /EXCLUDE:excludes.txt >nul
 xcopy /D /y /s plugins\scriptedwizard\resources\* %CB_OUTPUT_RESDIR%\templates\wizard /EXCLUDE:excludes.txt >nul
+echo wx-app-ash.cpp  >> excludes.txt
+echo wx-app-ash.h    >> excludes.txt
+echo wx-main-ash.cpp >> excludes.txt
+echo wx-main-ash.h   >> excludes.txt
+echo wx-main-s.cpp   >> excludes.txt
+echo wx-main-sh.cpp  >> excludes.txt
+echo wx-main-sh.h    >> excludes.txt
+echo wx_pch.h        >> excludes.txt
+echo wxwidgets.png   >> excludes.txt
+echo wxwidgets.cbp   >> excludes.txt
+echo wxwidgets.template      >> excludes.txt
+echo wxwidgets.cbp           >> excludes.txt
+echo wxwidgets_static.cbp    >> excludes.txt
+echo wxwidgets_u.cbp         >> excludes.txt
+echo wxwidgets_static_u.cbp  >> excludes.txt
 xcopy /D /y templates\common\* %CB_OUTPUT_RESDIR%\templates /EXCLUDE:excludes.txt > nul
 xcopy /D /y templates\win32\* %CB_OUTPUT_RESDIR%\templates /EXCLUDE:excludes.txt > nul
 xcopy /D /y templates\common\* %CB_DEVEL_RESDIR%\templates /EXCLUDE:excludes.txt > nul
