Index: src/plugins/abbreviations/abbreviations.cpp
===================================================================
--- src/plugins/abbreviations/abbreviations.cpp	(revision 9915)
+++ src/plugins/abbreviations/abbreviations.cpp	(working copy)
@@ -10,6 +10,8 @@
 #include <sdk.h>
 
 #ifndef CB_PRECOMP
+    #include <wx/textdlg.h>
+    
     #include <configurationpanel.h>
     #include <cbstyledtextctrl.h>
     #include <editorcolourset.h>
Index: src/plugins/abbreviations/abbreviationsconfigpanel.cpp
===================================================================
--- src/plugins/abbreviations/abbreviationsconfigpanel.cpp	(revision 9915)
+++ src/plugins/abbreviations/abbreviationsconfigpanel.cpp	(working copy)
@@ -7,20 +7,29 @@
  * $HeadURL$
  */
 
+// sdk.h must be the first include in the file to work as a PCH
 #include <sdk.h>
-#include "abbreviationsconfigpanel.h"
 
 #ifndef CB_PRECOMP
-    #include <wx/xrc/xmlres.h>
-    #include <wx/fontutil.h>
+    // wxWidgets non GUI headers in sdk_common.h
     #include <wx/choicdlg.h>
 
+    // wxWidgets GUI headers in sdk_common.h
+    #include <wx/combobox.h>
+    #include <wx/textdlg.h>
+    #include <wx/xrc/xmlres.h>
+
+    // CB SDK Headers in sdk_common.h
+    #include <editorcolourset.h>
     #include <configmanager.h>
-    #include <editorcolourset.h>
     #include <editormanager.h>
     #include <logmanager.h>
-#endif
+#endif // #ifndef CB_PRECOMP
 
+#include <wx/fontutil.h>
+
+#include "abbreviationsconfigpanel.h"
+
 #include "cbcolourmanager.h"
 
 BEGIN_EVENT_TABLE(AbbreviationsConfigPanel, cbConfigurationPanel)
Index: src/plugins/astyle/astyleplugin.cpp
===================================================================
--- src/plugins/astyle/astyleplugin.cpp	(revision 9915)
+++ src/plugins/astyle/astyleplugin.cpp	(working copy)
@@ -15,6 +15,7 @@
     #include <string>
     #include <vector>
 
+    #include <wx/menu.h>
     #include <wx/msgdlg.h>
     #include <wx/xrc/xmlres.h>
     #include <wx/fs_zip.h>
Index: src/plugins/codecompletion/cctest/cctest_frame.cpp
===================================================================
--- src/plugins/codecompletion/cctest/cctest_frame.cpp	(revision 9915)
+++ src/plugins/codecompletion/cctest/cctest_frame.cpp	(working copy)
@@ -6,12 +6,7 @@
  * $HeadURL$
  */
 
-#ifdef __BORLANDC__
-#pragma hdrstop
-#endif
-
 #ifndef WX_PRECOMP
-    #include <wx/wx.h>
     #include <wx/file.h>
 #endif
 
Index: src/plugins/codecompletion/selectincludefile.cpp
===================================================================
--- src/plugins/codecompletion/selectincludefile.cpp	(revision 9915)
+++ src/plugins/codecompletion/selectincludefile.cpp	(working copy)
@@ -10,6 +10,7 @@
 #include "prep.h"
 #include "selectincludefile.h"
 
+#include <wx/listbox.h>
 #include <wx/xrc/xmlres.h>
 #include <wx/arrstr.h>
 
Index: src/plugins/contrib/BrowseTracker/JumpTracker.cpp
===================================================================
--- src/plugins/contrib/BrowseTracker/JumpTracker.cpp	(revision 9915)
+++ src/plugins/contrib/BrowseTracker/JumpTracker.cpp	(working copy)
@@ -2,6 +2,7 @@
 //  JumpTracker.cpp
 // ----------------------------------------------------------------------------
 #include <sdk.h> // Code::Blocks SDK
+#include <wx/menu.h>
 #include <configurationpanel.h>
 #include <cbstyledtextctrl.h>
 #include <editormanager.h>
Index: src/plugins/contrib/Cscope/CscopePlugin.cpp
===================================================================
--- src/plugins/contrib/Cscope/CscopePlugin.cpp	(revision 9915)
+++ src/plugins/contrib/Cscope/CscopePlugin.cpp	(working copy)
@@ -1,4 +1,5 @@
 #include <sdk.h> // Code::Blocks SDK
+#include <wx/menu.h>
 #ifndef CB_PRECOMP
     #include <wx/process.h>
 
Index: src/plugins/contrib/FileManager/FileExplorer.cpp
===================================================================
--- src/plugins/contrib/FileManager/FileExplorer.cpp	(revision 9915)
+++ src/plugins/contrib/FileManager/FileExplorer.cpp	(working copy)
@@ -2,8 +2,11 @@
 #include <wx/dir.h>
 #include <wx/filename.h>
 #include <wx/aui/aui.h>
+#include <wx/stattext.h>
+#include <wx/menu.h>
+#include <wx/textdlg.h>
+#include <wx/dirdlg.h>
 
-
 #include <sdk.h>
 #ifndef CB_PRECOMP
     #include <wx/dnd.h>
Index: src/plugins/contrib/FileManager/FileExplorerSettings.cpp
===================================================================
--- src/plugins/contrib/FileManager/FileExplorerSettings.cpp	(revision 9915)
+++ src/plugins/contrib/FileManager/FileExplorerSettings.cpp	(working copy)
@@ -1,5 +1,8 @@
 #include "FileExplorerSettings.h"
 
+#include <wx/dirdlg.h>
+#include <wx/listbox.h>
+
 #include <sdk.h>
 #ifndef CB_PRECOMP
   #include <wx/xrc/xmlres.h>
Index: src/plugins/contrib/FileManager/FileExplorerUpdater.cpp
===================================================================
--- src/plugins/contrib/FileManager/FileExplorerUpdater.cpp	(revision 9915)
+++ src/plugins/contrib/FileManager/FileExplorerUpdater.cpp	(working copy)
@@ -1,3 +1,7 @@
+#include <wx/dir.h>
+#include <wx/timer.h>
+#include <wx/txtstrm.h>
+
 #include "FileExplorerUpdater.h"
 #include "FileExplorer.h"
 #include "se_globals.h"
Index: src/plugins/contrib/FileManager/FileManager.cpp
===================================================================
--- src/plugins/contrib/FileManager/FileManager.cpp	(revision 9915)
+++ src/plugins/contrib/FileManager/FileManager.cpp	(working copy)
@@ -1,4 +1,5 @@
 #include <wx/aui/aui.h>
+#include <wx/menu.h>
 #include <sdk.h> // Code::Blocks SDK
 #ifndef CB_PRECOMP
     #include <cbauibook.h>
Index: src/plugins/contrib/FileManager/directorymonitor.cpp
===================================================================
--- src/plugins/contrib/FileManager/directorymonitor.cpp	(revision 9915)
+++ src/plugins/contrib/FileManager/directorymonitor.cpp	(working copy)
@@ -1,8 +1,11 @@
+#include <vector>
+#include <iostream>
+
+#include <manager.h>
+
 #include "directorymonitor.h"
-#include <vector>
 #include "se_globals.h"
 
-#include <iostream>
 
 DEFINE_EVENT_TYPE(wxEVT_MONITOR_NOTIFY)
 DEFINE_EVENT_TYPE(wxEVT_MONITOR_NOTIFY2)
Index: src/plugins/contrib/FileManager/se_globals.cpp
===================================================================
--- src/plugins/contrib/FileManager/se_globals.cpp	(revision 9915)
+++ src/plugins/contrib/FileManager/se_globals.cpp	(working copy)
@@ -1,3 +1,6 @@
+#include <editorbase.h>
+#include <editormanager.h>
+
 #include "se_globals.h"
 
 wxString GetParentDir(const wxString &path)
Index: src/plugins/contrib/NassiShneiderman/GraphBricks.cpp
===================================================================
--- src/plugins/contrib/NassiShneiderman/GraphBricks.cpp	(revision 9915)
+++ src/plugins/contrib/NassiShneiderman/GraphBricks.cpp	(working copy)
@@ -1,3 +1,7 @@
+#include <wx/brush.h>
+#include <wx/dc.h>
+#include <wx/pen.h>
+
 #include "GraphBricks.h"
 #include "TextGraph.h"
 #include "bricks.h"
Index: src/plugins/contrib/NassiShneiderman/InsertBrickTask.cpp
===================================================================
--- src/plugins/contrib/NassiShneiderman/InsertBrickTask.cpp	(revision 9915)
+++ src/plugins/contrib/NassiShneiderman/InsertBrickTask.cpp	(working copy)
@@ -1,3 +1,6 @@
+#include <wx/event.h>
+#include <wx/image.h>
+
 #include "InsertBrickTask.h"
 #include "GraphBricks.h"
 #include "RedLineDrawlet.h"
Index: src/plugins/contrib/NassiShneiderman/NassiDiagramWindow.cpp
===================================================================
--- src/plugins/contrib/NassiShneiderman/NassiDiagramWindow.cpp	(revision 9915)
+++ src/plugins/contrib/NassiShneiderman/NassiDiagramWindow.cpp	(working copy)
@@ -1,21 +1,8 @@
-#ifdef __GNUG__
-// #pragma implementation
-#endif
 
-// For compilers that support precompilation, includes "wx/wx.h".
-#include <wx/wxprec.h>
-
-#ifdef __BORLANDC__
-    #pragma hdrstop
-#endif
-
-#ifndef WX_PRECOMP
-    #include <wx/wx.h>
-#endif
-
 #include <wx/bitmap.h>
 #include <wx/image.h>
 #include <wx/dcbuffer.h>
+#include <wx/settings.h>
 
 #include "NassiView.h"
 #include "NassiDiagramWindow.h"
Index: src/plugins/contrib/NassiShneiderman/NassiDropTarget.cpp
===================================================================
--- src/plugins/contrib/NassiShneiderman/NassiDropTarget.cpp	(revision 9915)
+++ src/plugins/contrib/NassiShneiderman/NassiDropTarget.cpp	(working copy)
@@ -1,10 +1,7 @@
 
-#ifdef __GNUG__
-// #pragma interface
-#endif
 
+#include <wx/msgdlg.h> 
 //#include <wx/dcmemory.h>
-//#include <wx/msgdlg.h>
 //#include <wx/mstream.h>
 //#include <wx/txtstrm.h>
 
Index: src/plugins/contrib/NassiShneiderman/NassiEditorPanel.cpp
===================================================================
--- src/plugins/contrib/NassiShneiderman/NassiEditorPanel.cpp	(revision 9915)
+++ src/plugins/contrib/NassiShneiderman/NassiEditorPanel.cpp	(working copy)
@@ -1,3 +1,4 @@
+#include <wx/sizer.h>
 
 #include <manager.h>
 #include <editormanager.h>
Index: src/plugins/contrib/NassiShneiderman/NassiPlugin.cpp
===================================================================
--- src/plugins/contrib/NassiShneiderman/NassiPlugin.cpp	(revision 9915)
+++ src/plugins/contrib/NassiShneiderman/NassiPlugin.cpp	(working copy)
@@ -1,11 +1,13 @@
-#include <wx/sstream.h>
-#include <wx/txtstrm.h>
-
-
 #include <sdk.h> // Code::Blocks SDK
 #ifndef CB_PRECOMP
+    #include <wx/menu.h>
+    #include <wx/toolbar.h>
+    
     #include <cbeditor.h>
 #endif
+#include <wx/sstream.h>
+#include <wx/txtstrm.h>
+
 #include <configurationpanel.h>
 #include <cbstyledtextctrl.h>
 #include <filefilters.h>
Index: src/plugins/contrib/NassiShneiderman/NassiView.cpp
===================================================================
--- src/plugins/contrib/NassiShneiderman/NassiView.cpp	(revision 9915)
+++ src/plugins/contrib/NassiShneiderman/NassiView.cpp	(working copy)
@@ -1,3 +1,9 @@
+#include <wx/dcclient.h>
+#include <wx/filedlg.h>
+#include <wx/dcmemory.h>
+#include <wx/icon.h>
+#include <wx/image.h> // Fixes Weird Compiler error: invalid conversion from 'const char**' to 'int' [-fpermissive]
+
 #include "NassiView.h"
 #include "NassiFileContent.h"
 #include "NassiDiagramWindow.h"
Index: src/plugins/contrib/NassiShneiderman/PasteTask.cpp
===================================================================
--- src/plugins/contrib/NassiShneiderman/PasteTask.cpp	(revision 9915)
+++ src/plugins/contrib/NassiShneiderman/PasteTask.cpp	(working copy)
@@ -1,3 +1,7 @@
+#include <wx/cursor.h>
+#include <wx/event.h>
+#include <wx/image.h>
+
 #include "PasteTask.h"
 #include "GraphBricks.h"
 #include "RedLineDrawlet.h"
Index: src/plugins/contrib/NassiShneiderman/RedHatchDrawlet.cpp
===================================================================
--- src/plugins/contrib/NassiShneiderman/RedHatchDrawlet.cpp	(revision 9915)
+++ src/plugins/contrib/NassiShneiderman/RedHatchDrawlet.cpp	(working copy)
@@ -1,3 +1,6 @@
+#include <wx/dc.h>
+#include <wx/version.h>
+
 #include "RedHatchDrawlet.h"
 
 RedHatchDrawlet::RedHatchDrawlet(wxRect rect):
Index: src/plugins/contrib/NassiShneiderman/RedLineDrawlet.cpp
===================================================================
--- src/plugins/contrib/NassiShneiderman/RedLineDrawlet.cpp	(revision 9915)
+++ src/plugins/contrib/NassiShneiderman/RedLineDrawlet.cpp	(working copy)
@@ -1,3 +1,6 @@
+#include <wx/dc.h>
+#include <wx/version.h>
+
 #include "RedLineDrawlet.h"
 
 RedLineDrawlet::RedLineDrawlet(wxPoint pos, wxCoord length, bool vertical):
Index: src/plugins/contrib/NassiShneiderman/TextCtrl.cpp
===================================================================
--- src/plugins/contrib/NassiShneiderman/TextCtrl.cpp	(revision 9915)
+++ src/plugins/contrib/NassiShneiderman/TextCtrl.cpp	(working copy)
@@ -1,3 +1,5 @@
+#include <wx/dcclient.h>
+
 #include "TextCtrl.h"
 #include "NassiDiagramWindow.h"
 
Index: src/plugins/contrib/NassiShneiderman/TextGraph.cpp
===================================================================
--- src/plugins/contrib/NassiShneiderman/TextGraph.cpp	(revision 9915)
+++ src/plugins/contrib/NassiShneiderman/TextGraph.cpp	(working copy)
@@ -1,3 +1,5 @@
+#include <wx/dc.h>
+
 #include "TextGraph.h"
 
 #include "bricks.h"
Index: src/plugins/contrib/NassiShneiderman/bricks.cpp
===================================================================
--- src/plugins/contrib/NassiShneiderman/bricks.cpp	(revision 9915)
+++ src/plugins/contrib/NassiShneiderman/bricks.cpp	(working copy)
@@ -1,26 +1,11 @@
-#ifdef __GNUG__
-// #pragma implementation
-#endif
 
-// For compilers that support precompilation, includes "wx/wx.h".
-#include <wx/wxprec.h>
-
-#ifdef __BORLANDC__
-#pragma hdrstop
-#endif
-
-#ifndef WX_PRECOMP
-#include <wx/wx.h>
-#endif
-
-#include <wx/arrimpl.cpp> // this is a magic incantation which must be done!
-
 #include <wx/txtstrm.h>
 #include <wx/arrstr.h>
 
 #include "bricks.h"
 #include "NassiBrickVisitor.h"
 
+// #include <wx/arrimpl.cpp> // this is a magic incantation which must be done!
 
 enum {
     NASSI_BRICK_SELECT      =   0,
Index: src/plugins/contrib/NassiShneiderman/cbEditorPanel.cpp
===================================================================
--- src/plugins/contrib/NassiShneiderman/cbEditorPanel.cpp	(revision 9915)
+++ src/plugins/contrib/NassiShneiderman/cbEditorPanel.cpp	(working copy)
@@ -20,6 +20,7 @@
 * $HeadURL$
 */
 
+#include <wx/filedlg.h>
 #include <wx/cmdproc.h>
 #include <configmanager.h>
 #include "cbEditorPanel.h"
Index: src/plugins/contrib/NassiShneiderman/commands.cpp
===================================================================
--- src/plugins/contrib/NassiShneiderman/commands.cpp	(revision 9915)
+++ src/plugins/contrib/NassiShneiderman/commands.cpp	(working copy)
@@ -1,18 +1,5 @@
-#ifdef __GNUG__
-// #pragma implementation
-#endif
 
-// For compilers that support precompilation, includes "wx/wx.h".
-#include <wx/wxprec.h>
-
-#ifdef __BORLANDC__
-#pragma hdrstop
-#endif
-
-#ifndef WX_PRECOMP
-#include <wx/wx.h>
-#endif
-
+#include <wx/intl.h>
 #include <wx/txtstrm.h>
 #include "commands.h"
 
Index: src/plugins/contrib/NassiShneiderman/parseactions.cpp
===================================================================
--- src/plugins/contrib/NassiShneiderman/parseactions.cpp	(revision 9915)
+++ src/plugins/contrib/NassiShneiderman/parseactions.cpp	(working copy)
@@ -1,18 +1,5 @@
-#ifdef __GNUG__
-// #pragma implementation
-#endif
+#include <wx/intl.h>
 
-// For compilers that support precompilation, includes "wx/wx.h".
-#include <wx/wxprec.h>
-
-#ifdef __BORLANDC__
-#pragma hdrstop
-#endif
-
-#ifndef WX_PRECOMP
-#include <wx/wx.h>
-#endif
-
 #include "parseactions.h"
 #include "bricks.h"
 
Index: src/plugins/contrib/ToolsPlus/CmdConfigDialog.cpp
===================================================================
--- src/plugins/contrib/ToolsPlus/CmdConfigDialog.cpp	(revision 9915)
+++ src/plugins/contrib/ToolsPlus/CmdConfigDialog.cpp	(working copy)
@@ -5,16 +5,9 @@
 // PLEASE DO "NOT" EDIT THIS FILE!
 ///////////////////////////////////////////////////////////////////////////
 
-#include "wx/wxprec.h"
+#include <sdk.h>
+#include <wx/filedlg.h>
 
-#ifdef __BORLANDC__
-#pragma hdrstop
-#endif //__BORLANDC__
-
-#ifndef WX_PRECOMP
-#include <wx/wx.h>
-#endif //WX_PRECOMP
-
 #include "ToolsPlus.h"
 #include "CmdConfigDialog.h"
 
Index: src/plugins/contrib/ToolsPlus/PipedProcessCtrl.cpp
===================================================================
--- src/plugins/contrib/ToolsPlus/PipedProcessCtrl.cpp	(revision 9915)
+++ src/plugins/contrib/ToolsPlus/PipedProcessCtrl.cpp	(working copy)
@@ -1,9 +1,24 @@
-#include <wx/notebook.h>
-#include <wx/textctrl.h>
-#include <wx/regex.h>
+// sdk.h must be the first include in the file to work as a PCH
+#include <sdk.h>
+
+#ifndef CB_PRECOMP
+    // wxWidgets non GUI headers in sdk_common.h
+    #include <wx/filename.h>
+    #include <wx/regex.h>
+
+    // wxWidgets GUI headers in sdk_common.h
+    #include <wx/notebook.h>
+    #include <wx/textctrl.h>
+
+    // CB SDK Headers in sdk_common.h
+    #include <globals.h>
+    #include <cbeditor.h>
+    #include <manager.h>
+    #include <configmanager.h>
+    #include <editormanager.h>
+#endif // #ifndef CB_PRECOMP
+
 #include "PipedProcessCtrl.h"
-#include <globals.h>
-#include <cbeditor.h>
 
 ////////////////////////////////////// PipedProcessCtrl /////////////////////////////////////////////
 #define PP_ERROR_STYLE 1
Index: src/plugins/contrib/ToolsPlus/ShellCtrlBase.cpp
===================================================================
--- src/plugins/contrib/ToolsPlus/ShellCtrlBase.cpp	(revision 9915)
+++ src/plugins/contrib/ToolsPlus/ShellCtrlBase.cpp	(working copy)
@@ -1,8 +1,20 @@
-#include <wx/notebook.h>
-#include <wx/textctrl.h>
-#include <wx/regex.h>
+// sdk.h must be the first include in the file to work as a PCH
+#include <sdk.h>
+
+#ifndef CB_PRECOMP
+    // wxWidgets non GUI headers in sdk_common.h
+    #include <wx/regex.h>
+
+    // wxWidgets GUI headers in sdk_common.h
+    #include <wx/menu.h>
+    #include <wx/notebook.h>
+    #include <wx/textctrl.h>
+
+    // CB SDK Headers in sdk_common.h
+    #include <globals.h>
+#endif // #ifndef CB_PRECOMP
+
 #include "ShellCtrlBase.h"
-#include <globals.h>
 
 // The global instance of the shell registry
 ShellRegistry& GlobalShellRegistry()
Index: src/plugins/contrib/ToolsPlus/ToolsPlus.cpp
===================================================================
--- src/plugins/contrib/ToolsPlus/ToolsPlus.cpp	(revision 9915)
+++ src/plugins/contrib/ToolsPlus/ToolsPlus.cpp	(working copy)
@@ -1,3 +1,28 @@
+// sdk.h must be the first include in the file to work as a PCH
+#include <sdk.h>
+
+#if !defined(WX_PRECOMP)
+    // wxWidgets non GUI headers in wx/wx.h; but, not in sdk_common.h
+
+    // wxWidgets GUI headers in wx/wx.h; but, not in sdk_common.h
+    #include <wx/dirdlg.h>
+#endif // #if !defined(WX_PRECOMP)
+
+#ifndef CB_PRECOMP
+    // wxWidgets non GUI headers in sdk_common.h
+
+    // wxWidgets GUI headers in sdk_common.h
+    #include <wx/filedlg.h>
+    #include <wx/menu.h>
+    #include <wx/textdlg.h>
+
+    // CB SDK Headers in sdk_common.h
+    #include <editorbase.h>
+    #include <cbproject.h>
+    #include <editormanager.h>
+    #include <macrosmanager.h>
+#endif // #ifndef CB_PRECOMP
+
 #include "ToolsPlus.h"
 #include "se_globals.h"
 
Index: src/plugins/contrib/ToolsPlus/se_globals.cpp
===================================================================
--- src/plugins/contrib/ToolsPlus/se_globals.cpp	(revision 9915)
+++ src/plugins/contrib/ToolsPlus/se_globals.cpp	(working copy)
@@ -1,3 +1,12 @@
+// sdk.h must be the first include in the file to work as a PCH
+#include <sdk.h>
+
+#ifndef CB_PRECOMP
+    // CB SDK Headers in sdk_common.h
+    #include <editorbase.h>
+    #include <editormanager.h>
+#endif // #ifndef CB_PRECOMP
+
 #include "se_globals.h"
 
 wxString GetParentDir(const wxString &path)
Index: src/plugins/contrib/source_exporter/exporter.cpp
===================================================================
--- src/plugins/contrib/source_exporter/exporter.cpp	(revision 9915)
+++ src/plugins/contrib/source_exporter/exporter.cpp	(working copy)
@@ -7,6 +7,8 @@
  **************************************************************/
 
 #include <wx/frame.h> // GetMenuBar
+#include <wx/menu.h>
+#include <wx/filedlg.h>
 #include "exporter.h"
 #include <configmanager.h>
 #include <manager.h>
Index: src/plugins/contrib/wxSmith/wxseditor.cpp
===================================================================
--- src/plugins/contrib/wxSmith/wxseditor.cpp	(revision 9915)
+++ src/plugins/contrib/wxSmith/wxseditor.cpp	(working copy)
@@ -24,7 +24,6 @@
 #include "wxsresource.h"
 #include "wxsmith.h"
 
-#include <wx/wx.h>
 
 wxsEditor::wxsEditor(wxWindow* parent, const wxString& title,wxsResource* Resource):
     EditorBase(parent,title),
Index: src/plugins/contrib/wxSmithAui/wxAuiToolBar/wxSmithAuiToolBar.cpp
===================================================================
--- src/plugins/contrib/wxSmithAui/wxAuiToolBar/wxSmithAuiToolBar.cpp	(revision 9915)
+++ src/plugins/contrib/wxSmithAui/wxAuiToolBar/wxSmithAuiToolBar.cpp	(working copy)
@@ -16,6 +16,8 @@
 * along with wxSmithAui. If not, see <http://www.gnu.org/licenses/>.
 */
 
+#include <wx/sizer.h>
+
 #include "wxSmithAuiToolBar.h"
 
 #if wxCHECK_VERSION(2,8,9)
Index: src/plugins/occurrenceshighlighting/highlighter.cpp
===================================================================
--- src/plugins/occurrenceshighlighting/highlighter.cpp	(revision 9915)
+++ src/plugins/occurrenceshighlighting/highlighter.cpp	(working copy)
@@ -7,8 +7,6 @@
  * $HeadURL$
  */
 
-#include "highlighter.h"
-
 #include <sdk.h> // Code::Blocks SDK
 
 #ifndef CB_PRECOMP
@@ -21,6 +19,8 @@
 #include <cbstyledtextctrl.h>
 #include <cbcolourmanager.h>
 
+#include "highlighter.h"
+
 HighlightedEditorPositions::HighlightedEditorPositions():
     linea(-1),
     lineb(-1),
Index: src/plugins/occurrenceshighlighting/occurrenceshighlighting.cpp
===================================================================
--- src/plugins/occurrenceshighlighting/occurrenceshighlighting.cpp	(revision 9915)
+++ src/plugins/occurrenceshighlighting/occurrenceshighlighting.cpp	(working copy)
@@ -12,6 +12,7 @@
 #ifndef CB_PRECOMP
     #include <algorithm>
     #include <wx/listctrl.h>
+    #include <wx/menu.h>
     #include <configmanager.h>
     #include <cbeditor.h>
     #include <editormanager.h>
Index: src/plugins/todo/todolistview.cpp
===================================================================
--- src/plugins/todo/todolistview.cpp	(revision 9915)
+++ src/plugins/todo/todolistview.cpp	(working copy)
@@ -23,6 +23,7 @@
 
     #include "cbeditor.h"
     #include "cbproject.h"
+    #include "configmanager.h"
     #include "editormanager.h"
     #include "filemanager.h"
     #include "globals.h"
Index: src/sdk/editorlexerloader.cpp
===================================================================
--- src/sdk/editorlexerloader.cpp	(revision 9915)
+++ src/sdk/editorlexerloader.cpp	(working copy)
@@ -7,20 +7,26 @@
  * $HeadURL$
  */
 
+// sdk_precomp.h must be the first include in the file to work as a PCH
 #include "sdk_precomp.h"
 
 #ifndef CB_PRECOMP
+    // wxWidgets non GUI headers in sdk_common.h
+    #include <wx/dynarray.h>
+    #include <wx/regex.h>
+    
+    // CB SDK Headers in sdk_common.h
+    #include "wx/wxscintilla.h"
     #include "globals.h"
     #include "manager.h"
+    #include "filemanager.h"
     #include "logmanager.h"
-    #include <wx/dynarray.h>
-    #include <wx/regex.h>
-    #include <wx/wxscintilla.h>
 #endif
 
 #include "editorcolourset.h"
+#include "tinyxml/tinyxml.h"
+
 #include "editorlexerloader.h"
-#include "filemanager.h"
 
 EditorLexerLoader::EditorLexerLoader(EditorColourSet* target)
     : m_pTarget(target)
Index: src/sdk/filemanager.cpp
===================================================================
--- src/sdk/filemanager.cpp	(revision 9915)
+++ src/sdk/filemanager.cpp	(working copy)
@@ -7,9 +7,11 @@
  * $HeadURL$
  */
 
+// sdk_precomp.h must be the first include in the file to work as a PCH
 #include "sdk_precomp.h"
 
 #ifndef CB_PRECOMP
+    #include "configmanager.h"
     #include "filemanager.h"
     #include "safedelete.h"
     #include "cbeditor.h"
Index: src/src/debugger_interface_creator.cpp
===================================================================
--- src/src/debugger_interface_creator.cpp	(revision 9915)
+++ src/src/debugger_interface_creator.cpp	(working copy)
@@ -2,13 +2,17 @@
  * This file is part of the Code::Blocks IDE and licensed under the GNU Lesser General Public License, version 3
  * http://www.gnu.org/licenses/gpl-3.0.html
  */
-#include "debugger_interface_creator.h"
 
+// sdk.h must be the first include in the file to work as a PCH
+#include <sdk.h>
+
 #ifndef CB_PRECOMP
     #include "cbeditor.h"
     #include "editormanager.h"
 #endif
 
+#include "debugger_interface_creator.h"
+
 #include "backtracedlg.h"
 #include "breakpointsdlg.h"
 #include "cbstyledtextctrl.h"
Index: src/src/find_replace.cpp
===================================================================
--- src/src/find_replace.cpp	(revision 9915)
+++ src/src/find_replace.cpp	(working copy)
@@ -1,28 +1,35 @@
-#include "sdk.h"
+// sdk.h must be the first include in the file to work as a PCH
+#include <sdk.h>
 
-#include "find_replace.h"
-
 #ifndef CB_PRECOMP
+    // wxWidgets non GUI headers in sdk_common.h
     #include <wx/dir.h>
     #include <wx/regex.h>
 
-    #include "cbauibook.h"
-    #include "cbeditor.h"
-    #include "cbproject.h"
-    #include "editormanager.h"
-    #include "infowindow.h"
-    #include "projectfile.h"
-#endif
+    // wxWidgets GUI headers in sdk_common.h
 
+    // CB SDK Headers in sdk_common.h
+    #include <cbeditor.h>
+    #include <projectfile.h>
+    #include <cbproject.h>
+    #include <cbauibook.h>
+    #include <configmanager.h>
+    #include <editormanager.h>
+    #include <infowindow.h>
+#endif // #ifndef CB_PRECOMP
+
 #include <wx/progdlg.h>
 
-#include "annoyingdialog.h"
-#include "cbstyledtextctrl.h"
-#include "confirmreplacedlg.h"
-#include "encodingdetector.h"
-#include "findreplacedlg.h"
-#include "searchresultslog.h"
+#include <annoyingdialog.h>
+#include <cbstyledtextctrl.h>
+#include <confirmreplacedlg.h>
+#include <encodingdetector.h>
+#include <findreplacedlg.h>
+#include <searchresultslog.h>
 
+#include "find_replace.h"
+
+
 struct cbFindReplaceData
 {
     int start;
Index: src/src/main.cpp
===================================================================
--- src/src/main.cpp	(revision 9915)
+++ src/src/main.cpp	(working copy)
@@ -7,20 +7,74 @@
  * $HeadURL$
  */
 
+// sdk.h must be the first include in the file to work as a PCH
 #include <sdk.h>
 
+#if !defined(WX_PRECOMP)
+    // wxWidgets non GUI headers in wx/wx.h; but, not in sdk_common.h
+
+    // wxWidgets GUI headers in wx/wx.h; but, not in sdk_common.h
+    #include <wx/gdicmn.h>
+#endif // #if !defined(WX_PRECOMP)
+
+#ifndef CB_PRECOMP
+    // wxWidgets non GUI headers in sdk_common.h
+    #include <wx/filename.h>
+
+    // wxWidgets GUI headers in sdk_common.h
+    #include <wx/dcclient.h>
+    #include <wx/filedlg.h>
+    #include <wx/textdlg.h>
+    #include <wx/xrc/xmlres.h>
+
+    // CB SDK Headers in sdk_common.h
+    #include <editorcolourset.h>
+    #include <globals.h>
+    #include <sdk_events.h>
+    #include <cbexception.h>
+    #include <cbplugin.h>
+    #include <cbproject.h>
+    #include <cbworkspace.h>
+    #include <cbauibook.h>
+    #include <configmanager.h>
+    #include <editormanager.h>
+    #include <logmanager.h>
+    #include <projectmanager.h>
+    #include <scriptingmanager.h>
+    #include <toolsmanager.h>
+    #include <templatemanager.h>
+    #include <pluginmanager.h>
+    #include <personalitymanager.h>
+    #include <uservarmanager.h>
+    #include <infowindow.h>
+#endif // #ifndef CB_PRECOMP
+
+#include <wx/display.h>
+#include <wx/dnd.h>
+#include <wx/fileconf.h>
+#include <wx/printdlg.h>
+#include <wx/sstream.h>
+#include <wx/tipdlg.h>
+#include <wx/tokenzr.h>
+
+#include <annoyingdialog.h>
+#include <cbcolourmanager.h>
+#include <cbstatusbar.h>
+#include <cbstyledtextctrl.h>
+#include <ccmanager.h>
+#include <debuggermanager.h>
+#include <filefilters.h>
+#include <loggers.h>
+
 #include "app.h"
 #include "appglobals.h"
 #include "batchbuild.h"
-#include "cbauibook.h"
-#include "cbstyledtextctrl.h"
 #include "compilersettingsdlg.h"
 #include "debuggersettingsdlg.h"
 #include "dlgabout.h"
 #include "dlgaboutplugin.h"
 #include "environmentsettingsdlg.h"
 #include "infopane.h"
-#include "infowindow.h"
 #include "main.h"
 #include "notebookstyles.h"
 #include "printdlg.h"
@@ -28,46 +82,8 @@
 #include "scriptingsettingsdlg.h"
 #include "startherepage.h"
 #include "switcherdlg.h"
-#include "cbstatusbar.h"
-#include "loggers.h"
-
-#include <wx/display.h>
-#include <wx/dnd.h>
-#include <wx/fileconf.h>
-#include <wx/filename.h>
-#include <wx/gdicmn.h>
-#include <wx/printdlg.h>
-#include <wx/sstream.h>
-#include <wx/tipdlg.h>
-#include <wx/tokenzr.h>
-#include <wx/xrc/xmlres.h>
-
-#include <annoyingdialog.h>
-#include <cbexception.h>
-#include <cbplugin.h>
-#include <cbproject.h>
-#include <cbworkspace.h>
-#include <ccmanager.h>
-#include <configmanager.h>
-#include <debuggermanager.h>
-#include <editorcolourset.h>
-#include <editormanager.h>
-#include <filefilters.h>
-#include <globals.h>
-#include <logmanager.h>
-#include <personalitymanager.h>
-#include <pluginmanager.h>
-#include <projectmanager.h>
-#include <scriptingmanager.h>
-#include <sdk_events.h>
-#include <templatemanager.h>
-#include <toolsmanager.h>
-#include <uservarmanager.h>
-
 #include "debugger_interface_creator.h"
 #include "debuggermenu.h"
-
-#include "cbcolourmanager.h"
 #include "editorconfigurationdlg.h"
 #include "projectmanagerui.h"
 
