Index: src/plugins/contrib/ToolsPlus/CmdConfigDialog.cpp
===================================================================
--- src/plugins/contrib/ToolsPlus/CmdConfigDialog.cpp	(revision 9915)
+++ src/plugins/contrib/ToolsPlus/CmdConfigDialog.cpp	(working copy)
@@ -5,16 +5,9 @@
 // PLEASE DO "NOT" EDIT THIS FILE!
 ///////////////////////////////////////////////////////////////////////////
 
-#include "wx/wxprec.h"
+#include <sdk.h>
+#include <wx/filedlg.h>
 
-#ifdef __BORLANDC__
-#pragma hdrstop
-#endif //__BORLANDC__
-
-#ifndef WX_PRECOMP
-#include <wx/wx.h>
-#endif //WX_PRECOMP
-
 #include "ToolsPlus.h"
 #include "CmdConfigDialog.h"
 
Index: src/plugins/contrib/ToolsPlus/PipedProcessCtrl.cpp
===================================================================
--- src/plugins/contrib/ToolsPlus/PipedProcessCtrl.cpp	(revision 9915)
+++ src/plugins/contrib/ToolsPlus/PipedProcessCtrl.cpp	(working copy)
@@ -1,9 +1,24 @@
-#include <wx/notebook.h>
-#include <wx/textctrl.h>
-#include <wx/regex.h>
+// sdk.h must be the first include in the file to work as a PCH
+#include <sdk.h>
+
+#ifndef CB_PRECOMP
+    // wxWidgets non GUI headers in sdk_common.h
+    #include <wx/filename.h>
+    #include <wx/regex.h>
+
+    // wxWidgets GUI headers in sdk_common.h
+    #include <wx/notebook.h>
+    #include <wx/textctrl.h>
+
+    // CB SDK Headers in sdk_common.h
+    #include <globals.h>
+    #include <cbeditor.h>
+    #include <manager.h>
+    #include <configmanager.h>
+    #include <editormanager.h>
+#endif // #ifndef CB_PRECOMP
+
 #include "PipedProcessCtrl.h"
-#include <globals.h>
-#include <cbeditor.h>
 
 ////////////////////////////////////// PipedProcessCtrl /////////////////////////////////////////////
 #define PP_ERROR_STYLE 1
Index: src/plugins/contrib/ToolsPlus/ShellCtrlBase.cpp
===================================================================
--- src/plugins/contrib/ToolsPlus/ShellCtrlBase.cpp	(revision 9915)
+++ src/plugins/contrib/ToolsPlus/ShellCtrlBase.cpp	(working copy)
@@ -1,8 +1,20 @@
-#include <wx/notebook.h>
-#include <wx/textctrl.h>
-#include <wx/regex.h>
+// sdk.h must be the first include in the file to work as a PCH
+#include <sdk.h>
+
+#ifndef CB_PRECOMP
+    // wxWidgets non GUI headers in sdk_common.h
+    #include <wx/regex.h>
+
+    // wxWidgets GUI headers in sdk_common.h
+    #include <wx/menu.h>
+    #include <wx/notebook.h>
+    #include <wx/textctrl.h>
+
+    // CB SDK Headers in sdk_common.h
+    #include <globals.h>
+#endif // #ifndef CB_PRECOMP
+
 #include "ShellCtrlBase.h"
-#include <globals.h>
 
 // The global instance of the shell registry
 ShellRegistry& GlobalShellRegistry()
Index: src/plugins/contrib/ToolsPlus/ToolsPlus.cpp
===================================================================
--- src/plugins/contrib/ToolsPlus/ToolsPlus.cpp	(revision 9915)
+++ src/plugins/contrib/ToolsPlus/ToolsPlus.cpp	(working copy)
@@ -1,3 +1,28 @@
+// sdk.h must be the first include in the file to work as a PCH
+#include <sdk.h>
+
+#if !defined(WX_PRECOMP)
+    // wxWidgets non GUI headers in wx/wx.h; but, not in sdk_common.h
+
+    // wxWidgets GUI headers in wx/wx.h; but, not in sdk_common.h
+    #include <wx/dirdlg.h>
+#endif // #if !defined(WX_PRECOMP)
+
+#ifndef CB_PRECOMP
+    // wxWidgets non GUI headers in sdk_common.h
+
+    // wxWidgets GUI headers in sdk_common.h
+    #include <wx/filedlg.h>
+    #include <wx/menu.h>
+    #include <wx/textdlg.h>
+
+    // CB SDK Headers in sdk_common.h
+    #include <editorbase.h>
+    #include <cbproject.h>
+    #include <editormanager.h>
+    #include <macrosmanager.h>
+#endif // #ifndef CB_PRECOMP
+
 #include "ToolsPlus.h"
 #include "se_globals.h"
 
Index: src/plugins/contrib/ToolsPlus/se_globals.cpp
===================================================================
--- src/plugins/contrib/ToolsPlus/se_globals.cpp	(revision 9915)
+++ src/plugins/contrib/ToolsPlus/se_globals.cpp	(working copy)
@@ -1,3 +1,12 @@
+// sdk.h must be the first include in the file to work as a PCH
+#include <sdk.h>
+
+#ifndef CB_PRECOMP
+    // CB SDK Headers in sdk_common.h
+    #include <editorbase.h>
+    #include <editormanager.h>
+#endif // #ifndef CB_PRECOMP
+
 #include "se_globals.h"
 
 wxString GetParentDir(const wxString &path)
