Index: src/sdk/editorlexerloader.cpp
===================================================================
--- src/sdk/editorlexerloader.cpp	(revision 9916)
+++ src/sdk/editorlexerloader.cpp	(working copy)
@@ -7,20 +7,26 @@
  * $HeadURL$
  */
 
+// sdk_precomp.h must be the first include in the file to work as a PCH
 #include "sdk_precomp.h"
 
 #ifndef CB_PRECOMP
+    // wxWidgets non GUI headers in sdk_common.h
+    #include <wx/dynarray.h>
+    #include <wx/regex.h>
+    
+    // CB SDK Headers in sdk_common.h
+    #include "wx/wxscintilla.h"
     #include "globals.h"
     #include "manager.h"
+    #include "filemanager.h"
     #include "logmanager.h"
-    #include <wx/dynarray.h>
-    #include <wx/regex.h>
-    #include <wx/wxscintilla.h>
 #endif
 
 #include "editorcolourset.h"
+#include "tinyxml/tinyxml.h"
+
 #include "editorlexerloader.h"
-#include "filemanager.h"
 
 EditorLexerLoader::EditorLexerLoader(EditorColourSet* target)
     : m_pTarget(target)
Index: src/sdk/filemanager.cpp
===================================================================
--- src/sdk/filemanager.cpp	(revision 9916)
+++ src/sdk/filemanager.cpp	(working copy)
@@ -10,6 +10,7 @@
 #include "sdk_precomp.h"
 
 #ifndef CB_PRECOMP
+    #include "configmanager.h"
     #include "filemanager.h"
     #include "safedelete.h"
     #include "cbeditor.h"
