Index: src/plugins/contrib/FileManager/FileExplorer.cpp
===================================================================
--- src/plugins/contrib/FileManager/FileExplorer.cpp	(revision 9908)
+++ src/plugins/contrib/FileManager/FileExplorer.cpp	(working copy)
@@ -2,8 +2,11 @@
 #include <wx/dir.h>
 #include <wx/filename.h>
 #include <wx/aui/aui.h>
+#include <wx/stattext.h>
+#include <wx/menu.h>
+#include <wx/textdlg.h>
+#include <wx/dirdlg.h>
 
-
 #include <sdk.h>
 #ifndef CB_PRECOMP
     #include <wx/dnd.h>
Index: src/plugins/contrib/FileManager/FileExplorerSettings.cpp
===================================================================
--- src/plugins/contrib/FileManager/FileExplorerSettings.cpp	(revision 9908)
+++ src/plugins/contrib/FileManager/FileExplorerSettings.cpp	(working copy)
@@ -1,5 +1,8 @@
 #include "FileExplorerSettings.h"
 
+#include <wx/dirdlg.h>
+#include <wx/listbox.h>
+
 #include <sdk.h>
 #ifndef CB_PRECOMP
   #include <wx/xrc/xmlres.h>
Index: src/plugins/contrib/FileManager/FileExplorerUpdater.cpp
===================================================================
--- src/plugins/contrib/FileManager/FileExplorerUpdater.cpp	(revision 9908)
+++ src/plugins/contrib/FileManager/FileExplorerUpdater.cpp	(working copy)
@@ -1,3 +1,7 @@
+#include <wx/dir.h>
+#include <wx/timer.h>
+#include <wx/txtstrm.h>
+
 #include "FileExplorerUpdater.h"
 #include "FileExplorer.h"
 #include "se_globals.h"
Index: src/plugins/contrib/FileManager/FileManager.cpp
===================================================================
--- src/plugins/contrib/FileManager/FileManager.cpp	(revision 9908)
+++ src/plugins/contrib/FileManager/FileManager.cpp	(working copy)
@@ -1,4 +1,5 @@
 #include <wx/aui/aui.h>
+#include <wx/menu.h>
 #include <sdk.h> // Code::Blocks SDK
 #ifndef CB_PRECOMP
     #include <cbauibook.h>
Index: src/plugins/contrib/FileManager/directorymonitor.cpp
===================================================================
--- src/plugins/contrib/FileManager/directorymonitor.cpp	(revision 9908)
+++ src/plugins/contrib/FileManager/directorymonitor.cpp	(working copy)
@@ -1,8 +1,11 @@
+#include <vector>
+#include <iostream>
+
+#include <manager.h>
+
 #include "directorymonitor.h"
-#include <vector>
 #include "se_globals.h"
 
-#include <iostream>
 
 DEFINE_EVENT_TYPE(wxEVT_MONITOR_NOTIFY)
 DEFINE_EVENT_TYPE(wxEVT_MONITOR_NOTIFY2)
Index: src/plugins/contrib/FileManager/se_globals.cpp
===================================================================
--- src/plugins/contrib/FileManager/se_globals.cpp	(revision 9908)
+++ src/plugins/contrib/FileManager/se_globals.cpp	(working copy)
@@ -1,3 +1,6 @@
+#include <editorbase.h>
+#include <editormanager.h>
+
 #include "se_globals.h"
 
 wxString GetParentDir(const wxString &path)
