Index: src/sdk/wxscintilla/include/wx/wxscintilla.h
===================================================================
--- src/sdk/wxscintilla/include/wx/wxscintilla.h	(revision 9823)
+++ src/sdk/wxscintilla/include/wx/wxscintilla.h	(working copy)
@@ -33,8 +33,18 @@
 #if wxCHECK_VERSION(2,9,2)
 #include <wx/versioninfo.h>
 #endif
+#if defined(BUILDING_PLUGIN) || defined(EXPORT_LIB)
+    #include "settings.h"
+#else
+    #if defined(__WXMSW__)
+        #define DLLIMPORT __declspec (dllexport)
+    #endif // defined(__WXMSW__)
+#endif // defined(BUILDING_PLUGIN) || defined(EXPORT_LIB)
+#define WXDLLIMPEXP_SCI DLLIMPORT
+
+#if 0
 /* C::B end */
 
 #ifdef WXMAKINGDLL_SCI
     #define WXDLLIMPEXP_SCI WXEXPORT
 #elif defined(WXUSINGDLL_SCI) || defined(WXUSINGDLL)
@@ -42,7 +51,10 @@
 #else // not making nor using DLL
     #define WXDLLIMPEXP_SCI
 #endif
+/* C::B begin */
+#endif // #if 0
+/* C::B end */
 
 class WXDLLIMPEXP_FWD_CORE wxScrollBar;
 
 // SWIG can't handle "#if" type of conditionals, only "#ifdef"
Index: src/sdk/wxscintilla/src/ScintillaWX.h
===================================================================
--- src/sdk/wxscintilla/src/ScintillaWX.h	(revision 9823)
+++ src/sdk/wxscintilla/src/ScintillaWX.h	(working copy)
@@ -53,6 +53,9 @@
 
 //----------------------------------------------------------------------
 
+/* C::B begin */
+#if 0
+/* C::B end */
 #ifdef WXMAKINGDLL_SCI
     #define WXDLLIMPEXP_SCI WXEXPORT
 #elif defined(WXUSINGDLL)
@@ -60,8 +63,11 @@
 #else // not making nor using DLL
     #define WXDLLIMPEXP_SCI
 #endif
+/* C::B begin */
+#endif // #if 0
+/* C::B end */
 
 
 #ifdef SCI_NAMESPACE
 	#ifndef SCI_NAMESPACE_PREFIX
 		#define SCI_NAMESPACE_PREFIX( x ) Scintilla::x
