Index: src/CodeBlocks.cbp
===================================================================
--- src/CodeBlocks.cbp	(revision 9858)
+++ src/CodeBlocks.cbp	(working copy)
@@ -733,7 +733,6 @@
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DcbDEBUG" />
 			<Add option="-DCB_PRECOMP" />
-			<Add option="-DWX_PRECOMP" />
 			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#WX.include)" />
 			<Add directory="$(#WX.lib)/gcc_dll$(WX_CFG)/msw$(WX_SUFFIX)" />
Index: src/plugins/contrib/wxSmith/wxSmith.cbp
===================================================================
--- src/plugins/contrib/wxSmith/wxSmith.cbp	(revision 9858)
+++ src/plugins/contrib/wxSmith/wxSmith.cbp	(working copy)
@@ -45,17 +45,18 @@
 			<Add option="-fmessage-length=0" />
 			<Add option="-fexceptions" />
 			<Add option="-Winvalid-pch" />
+			<Add option="-include sdk.h" />
 			<Add option='[[if (GetCompilerFactory().GetCompilerVersionString(_T(&quot;gcc&quot;)) &gt;= _T(&quot;4.0.0&quot;)) print(_T(&quot;-Wno-attributes&quot;));]]' />
 			<Add option="-DHAVE_W32API_H" />
 			<Add option="-D__WXMSW__" />
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DcbDEBUG" />
 			<Add option="-DCB_PRECOMP" />
-			<Add option="-DWX_PRECOMP" />
 			<Add option="-DBUILDING_PLUGIN" />
 			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)\msw$(WX_SUFFIX)" />
 			<Add directory="$(#WX.include)" />
+			<Add directory="..\..\..\.objs\include" />
 			<Add directory="..\..\..\include" />
 			<Add directory="..\..\..\sdk\wxpropgrid\include" />
 			<Add directory="..\..\..\sdk\wxscintilla\include" />
Index: src/plugins/contrib/wxSmith/wxseditor.cpp
===================================================================
--- src/plugins/contrib/wxSmith/wxseditor.cpp	(revision 9858)
+++ src/plugins/contrib/wxSmith/wxseditor.cpp	(working copy)
@@ -24,7 +24,6 @@
 #include "wxsresource.h"
 #include "wxsmith.h"
 
-#include <wx/wx.h>
 
 wxsEditor::wxsEditor(wxWindow* parent, const wxString& title,wxsResource* Resource):
     EditorBase(parent,title),
Index: src/plugins/contrib/wxSmith/wxwidgets/defitems/wxstoolbareditor.h
===================================================================
--- src/plugins/contrib/wxSmith/wxwidgets/defitems/wxstoolbareditor.h	(revision 9858)
+++ src/plugins/contrib/wxSmith/wxwidgets/defitems/wxstoolbareditor.h	(working copy)
@@ -23,7 +23,6 @@
 #ifndef WXSTOOLBAREDITOR_H
 #define WXSTOOLBAREDITOR_H
 
-#include <wx/wxprec.h>
 
 #include "wxstoolbar.h"
 
Index: src/plugins/contrib/wxSmith/wxwidgets/properties/wxsfontfaceeditordlg.h
===================================================================
--- src/plugins/contrib/wxSmith/wxwidgets/properties/wxsfontfaceeditordlg.h	(revision 9858)
+++ src/plugins/contrib/wxSmith/wxwidgets/properties/wxsfontfaceeditordlg.h	(working copy)
@@ -23,12 +23,7 @@
 #ifndef WXSFONTFACEEDITORDLG_H
 #define WXSFONTFACEEDITORDLG_H
 
-#include <wx/wxprec.h>
 
-#ifdef __BORLANDC__
-    #pragma hdrstop
-#endif
-
 //(*Headers(wxsFontFaceEditorDlg)
 #include <wx/sizer.h>
 #include <wx/textctrl.h>
Index: src/plugins/contrib/wxSmithAui/wxAuiManager/wxSmithAuiManager.h
===================================================================
--- src/plugins/contrib/wxSmithAui/wxAuiManager/wxSmithAuiManager.h	(revision 9858)
+++ src/plugins/contrib/wxSmithAui/wxAuiManager/wxSmithAuiManager.h	(working copy)
@@ -2,7 +2,7 @@
 #define WXSMITHAUIMANAGER_H
 
 #include <wx/aui/aui.h>
-#include <wx/wx.h>
+
 #include <wx/event.h>
 
 #include <prep.h>
Index: src/plugins/contrib/wxSmithAui/wxAuiToolBar/wxSmithAuiToolBar.h
===================================================================
--- src/plugins/contrib/wxSmithAui/wxAuiToolBar/wxSmithAuiToolBar.h	(revision 9858)
+++ src/plugins/contrib/wxSmithAui/wxAuiToolBar/wxSmithAuiToolBar.h	(working copy)
@@ -19,10 +19,11 @@
 #ifndef WXSMITHAUITOOLBAR_H
 #define WXSMITHAUITOOLBAR_H
 
-#include <wx/wx.h>
+#include <wx/version.h>
 
 #if wxCHECK_VERSION(2,8,9)
 
+#include <wx/bitmap.h> 
 #include <wx/aui/auibar.h>
 
 class wxSmithAuiToolBar : public wxAuiToolBar
Index: src/plugins/contrib/wxSmithAui/wxSmithAui.cbp
===================================================================
--- src/plugins/contrib/wxSmithAui/wxSmithAui.cbp	(revision 9858)
+++ src/plugins/contrib/wxSmithAui/wxSmithAui.cbp	(working copy)
@@ -20,6 +20,7 @@
 					<Add option="-ansi" />
 					<Add option="`wx-config --cflags`" />
 					<Add option="-Winvalid-pch" />
+					<Add option="-include sdk.h" />
 					<Add option="-fexceptions" />
 					<Add option="-fPIC" />
 				</Compiler>
@@ -49,6 +50,7 @@
 					<Add option="-fmessage-length=0" />
 					<Add option="-fexceptions" />
 					<Add option="-Winvalid-pch" />
+					<Add option="-include sdk.h" />
 					<Add option="-DHAVE_W32API_H" />
 					<Add option="-D__WXMSW__" />
 					<Add directory="$(#wx.include)" />
@@ -83,7 +85,7 @@
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DcbDEBUG" />
 			<Add option="-DCB_PRECOMP" />
-			<Add option="-DWX_PRECOMP" />
+			<Add directory="../../../.objs/include" />
 			<Add directory="../../../include" />
 			<Add directory="../../../sdk/wxscintilla/include" />
 			<Add directory="../../../sdk/wxpropgrid/include" />
Index: src/plugins/contrib/wxSmithAui/wxSmithAui.cpp
===================================================================
--- src/plugins/contrib/wxSmithAui/wxSmithAui.cpp	(revision 9858)
+++ src/plugins/contrib/wxSmithAui/wxSmithAui.cpp	(working copy)
@@ -1,4 +1,6 @@
+#ifndef SDK_H
 #include <sdk.h> // Code::Blocks SDK
+#endif // #ifndef SDK_H
 #include <configurationpanel.h>
 #include "wxSmithAui.h"
 
Index: src/plugins/contrib/wxSmithAui/wxSmithAui.h
===================================================================
--- src/plugins/contrib/wxSmithAui/wxSmithAui.h	(revision 9858)
+++ src/plugins/contrib/wxSmithAui/wxSmithAui.h	(working copy)
@@ -10,13 +10,7 @@
 #ifndef WXSMITHAUI_H_INCLUDED
 #define WXSMITHAUI_H_INCLUDED
 
-// For compilers that support precompilation, includes <wx/wx.h>
-#include <wx/wxprec.h>
 
-#ifndef WX_PRECOMP
-    #include <wx/wx.h>
-#endif
-
 #include <cbplugin.h> // for "class cbPlugin"
 
 class wxSmithAui : public cbPlugin
