Index: src/plugins/contrib/envvars/envvars.cbp
===================================================================
--- src/plugins/contrib/envvars/envvars.cbp	(revision 10130)
+++ src/plugins/contrib/envvars/envvars.cbp	(working copy)
@@ -14,9 +14,11 @@
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --multiple-instance -na -ns -nd -p debug" />
 				<Option host_application="..\..\..\devel\codeblocks.exe" />
+				<Option projectIncludeDirsRelation="2" />
 				<Compiler>
 					<Add option="-DCB_PRECOMP" />
 					<Add option="-DWX_PRECOMP" />
+					<Add directory="../../../.objs/include" />
 				</Compiler>
 			</Target>
 			<Target title="no_pch">
Index: src/plugins/contrib/envvars/envvars_wx30-unix.cbp
===================================================================
--- src/plugins/contrib/envvars/envvars_wx30-unix.cbp	(revision 10130)
+++ src/plugins/contrib/envvars/envvars_wx30-unix.cbp	(working copy)
@@ -14,6 +14,11 @@
 				<Option parameters="--debug-log --multiple-instance -ns -ni -p debug" />
 				<Option host_application="../../../devel30/codeblocks" />
 				<Option run_host_application_in_terminal="0" />
+				<Option projectIncludeDirsRelation="2" />
+				<Compiler>
+					<Add option="-DCB_PRECOMP" />
+					<Add directory="../../../.objs30/include" />
+				</Compiler>
 			</Target>
 			<Environment>
 				<Variable name="WX_CFG" value="" />
@@ -33,7 +38,6 @@
 			<Add option="-Winvalid-pch" />
 			<Add option="-fexceptions" />
 			<Add option="-fPIC" />
-			<Add option="-DCB_PRECOMP" />
 			<Add option="-DcbDEBUG" />
 			<Add directory="../../../include" />
 			<Add directory="../../../include/scripting/include" />
Index: src/plugins/contrib/envvars/envvars-unix.cbp
===================================================================
--- src/plugins/contrib/envvars/envvars-unix.cbp	(revision 10130)
+++ src/plugins/contrib/envvars/envvars-unix.cbp	(working copy)
@@ -41,6 +41,7 @@
 			<Add option="-fPIC" />
 			<Add option="-DCB_PRECOMP" />
 			<Add option="-DcbDEBUG" />
+			<Add directory="../../../.objs/include" />
 			<Add directory="../../../include" />
 			<Add directory="../../../include/scripting/include" />
 			<Add directory="../../../include/scripting/sqplus" />
