Index: src/src/debugger_interface_creator.cpp
===================================================================
--- src/src/debugger_interface_creator.cpp	(revision 9964)
+++ src/src/debugger_interface_creator.cpp	(working copy)
@@ -2,13 +2,17 @@
  * This file is part of the Code::Blocks IDE and licensed under the GNU Lesser General Public License, version 3
  * http://www.gnu.org/licenses/gpl-3.0.html
  */
-#include "debugger_interface_creator.h"
 
+// sdk.h must be the first include in the file to work as a PCH
+#include <sdk.h>
+
 #ifndef CB_PRECOMP
     #include "cbeditor.h"
     #include "editormanager.h"
 #endif
 
+#include "debugger_interface_creator.h"
+
 #include "backtracedlg.h"
 #include "breakpointsdlg.h"
 #include "cbstyledtextctrl.h"
