Index: src/plugins/contrib/wxSmith/wxSmith-unix.cbp
===================================================================
--- src/plugins/contrib/wxSmith/wxSmith-unix.cbp	(revision 10123)
+++ src/plugins/contrib/wxSmith/wxSmith-unix.cbp	(working copy)
@@ -50,11 +50,13 @@
 			<Add option="$(#CB_RELEASE_TYPE)" />
 			<Add option="`wx-config --version=2.8 --cflags`" />
 			<Add option="-Winvalid-pch" />
+			<Add option='-include &quot;sdk.h&quot;' />
 			<Add option="-fexceptions" />
 			<Add option="-fPIC" />
 			<Add option='[[if (GetCompilerFactory().GetCompilerVersionString(_T(&quot;gcc&quot;)) &gt;= _T(&quot;4.0.0&quot;)) print(_T(&quot;-Wno-attributes&quot;));]]' />
 			<Add option="-DcbDEBUG" />
 			<Add option="-DCB_PRECOMP" />
+			<Add directory="../../../.objs/include" />
 			<Add directory="../../../include" />
 			<Add directory="../../../sdk/wxscintilla/include" />
 			<Add directory="../../../sdk/wxpropgrid/include" />
Index: src/plugins/contrib/wxSmith/wxSmith_wx30-unix.cbp
===================================================================
--- src/plugins/contrib/wxSmith/wxSmith_wx30-unix.cbp	(revision 10123)
+++ src/plugins/contrib/wxSmith/wxSmith_wx30-unix.cbp	(working copy)
@@ -52,11 +52,13 @@
 			<Add option="$(#CB_RELEASE_TYPE)" />
 			<Add option="`$(WX_CONFIG) --cflags`" />
 			<Add option="-Winvalid-pch" />
+			<Add option='-include &quot;sdk.h&quot;' />
 			<Add option="-fexceptions" />
 			<Add option="-fPIC" />
 			<Add option='[[if (GetCompilerFactory().GetCompilerVersionString(_T(&quot;gcc&quot;)) &gt;= _T(&quot;4.0.0&quot;)) print(_T(&quot;-Wno-attributes&quot;));]]' />
 			<Add option="-DcbDEBUG" />
 			<Add option="-DCB_PRECOMP" />
+			<Add directory="../../../.objs30/include" />
 			<Add directory="../../../include" />
 			<Add directory="../../../sdk/wxscintilla/include" />
 			<Add directory="../../../include/scripting/include" />
