Index: src/CodeBlocks-unix.cbp
===================================================================
--- src/CodeBlocks-unix.cbp	(revision 10002)
+++ src/CodeBlocks-unix.cbp	(working copy)
@@ -752,6 +752,9 @@
 		<Unit filename="include/mozilla_chardet/JpCntx.h">
 			<Option target="sdk" />
 		</Unit>
+		<Unit filename="include/mozilla_chardet/mfbt/mozilla/Alignment.h">
+			<Option target="sdk" />
+		</Unit>
 		<Unit filename="include/mozilla_chardet/mfbt/mozilla/Assertions.h">
 			<Option target="sdk" />
 		</Unit>
@@ -767,6 +770,9 @@
 		<Unit filename="include/mozilla_chardet/mfbt/mozilla/NullPtr.h">
 			<Option target="sdk" />
 		</Unit>
+		<Unit filename="include/mozilla_chardet/mfbt/mozilla/TypedEnum.h">
+			<Option target="sdk" />
+		</Unit>
 		<Unit filename="include/mozilla_chardet/mfbt/mozilla/Types.h">
 			<Option target="sdk" />
 		</Unit>
Index: src/CodeBlocks.cbp
===================================================================
--- src/CodeBlocks.cbp	(revision 10002)
+++ src/CodeBlocks.cbp	(working copy)
@@ -995,6 +995,9 @@
 		<Unit filename="include/mozilla_chardet/JpCntx.h">
 			<Option target="sdk" />
 		</Unit>
+		<Unit filename="include/mozilla_chardet/mfbt/mozilla/Alignment.h">
+			<Option target="sdk" />
+		</Unit>
 		<Unit filename="include/mozilla_chardet/mfbt/mozilla/Assertions.h">
 			<Option target="sdk" />
 		</Unit>
@@ -1013,6 +1016,9 @@
 		<Unit filename="include/mozilla_chardet/mfbt/mozilla/StandardInteger.h">
 			<Option target="sdk" />
 		</Unit>
+		<Unit filename="include/mozilla_chardet/mfbt/mozilla/TypedEnum.h">
+			<Option target="sdk" />
+		</Unit>
 		<Unit filename="include/mozilla_chardet/mfbt/mozilla/Types.h">
 			<Option target="sdk" />
 		</Unit>
Index: src/CodeBlocks_wx30-unix.cbp
===================================================================
--- src/CodeBlocks_wx30-unix.cbp	(revision 10002)
+++ src/CodeBlocks_wx30-unix.cbp	(working copy)
@@ -795,6 +795,9 @@
 		<Unit filename="include/mozilla_chardet/JpCntx.h">
 			<Option target="sdk" />
 		</Unit>
+		<Unit filename="include/mozilla_chardet/mfbt/mozilla/Alignment.h">
+			<Option target="sdk" />
+		</Unit>
 		<Unit filename="include/mozilla_chardet/mfbt/mozilla/Assertions.h">
 			<Option target="sdk" />
 		</Unit>
@@ -810,6 +813,9 @@
 		<Unit filename="include/mozilla_chardet/mfbt/mozilla/NullPtr.h">
 			<Option target="sdk" />
 		</Unit>
+		<Unit filename="include/mozilla_chardet/mfbt/mozilla/TypedEnum.h">
+			<Option target="sdk" />
+		</Unit>
 		<Unit filename="include/mozilla_chardet/mfbt/mozilla/Types.h">
 			<Option target="sdk" />
 		</Unit>
Index: src/CodeBlocks_wx30.cbp
===================================================================
--- src/CodeBlocks_wx30.cbp	(revision 10002)
+++ src/CodeBlocks_wx30.cbp	(working copy)
@@ -991,6 +991,9 @@
 		<Unit filename="include/mozilla_chardet/JpCntx.h">
 			<Option target="sdk" />
 		</Unit>
+		<Unit filename="include/mozilla_chardet/mfbt/mozilla/Alignment.h">
+			<Option target="sdk" />
+		</Unit>
 		<Unit filename="include/mozilla_chardet/mfbt/mozilla/Assertions.h">
 			<Option target="sdk" />
 		</Unit>
@@ -1009,6 +1012,9 @@
 		<Unit filename="include/mozilla_chardet/mfbt/mozilla/StandardInteger.h">
 			<Option target="sdk" />
 		</Unit>
+		<Unit filename="include/mozilla_chardet/mfbt/mozilla/TypedEnum.h">
+			<Option target="sdk" />
+		</Unit>
 		<Unit filename="include/mozilla_chardet/mfbt/mozilla/Types.h">
 			<Option target="sdk" />
 		</Unit>
Index: src/CodeBlocks_wx30_64.cbp
===================================================================
--- src/CodeBlocks_wx30_64.cbp	(revision 10002)
+++ src/CodeBlocks_wx30_64.cbp	(working copy)
@@ -994,6 +994,9 @@
 		<Unit filename="include/mozilla_chardet/JpCntx.h">
 			<Option target="sdk" />
 		</Unit>
+		<Unit filename="include/mozilla_chardet/mfbt/mozilla/Alignment.h">
+			<Option target="sdk" />
+		</Unit>
 		<Unit filename="include/mozilla_chardet/mfbt/mozilla/Assertions.h">
 			<Option target="sdk" />
 		</Unit>
@@ -1012,6 +1015,9 @@
 		<Unit filename="include/mozilla_chardet/mfbt/mozilla/StandardInteger.h">
 			<Option target="sdk" />
 		</Unit>
+		<Unit filename="include/mozilla_chardet/mfbt/mozilla/TypedEnum.h">
+			<Option target="sdk" />
+		</Unit>
 		<Unit filename="include/mozilla_chardet/mfbt/mozilla/Types.h">
 			<Option target="sdk" />
 		</Unit>
