Index: src/include/xtra_res.h
===================================================================
--- src/include/xtra_res.h	(revision 9935)
+++ src/include/xtra_res.h	(working copy)
@@ -6,16 +6,11 @@
 #ifndef XTRA_RES_H
 #define XTRA_RES_H
 
-#include <wx/wxprec.h>
 #include <wx/xrc/xmlres.h>
 #include <wx/xrc/xh_dlg.h>
 #include <wx/toolbar.h>
 
-#ifdef __BORLANDC__
-    #pragma hdrstop
-#endif
 
-
 class wxXmlResourceHandler;
 
 class wxToolBarAddOnXmlHandler : public wxXmlResourceHandler
