Index: src/CodeBlocks.cbp
===================================================================
--- src/CodeBlocks.cbp	(revision 9849)
+++ src/CodeBlocks.cbp	(working copy)
@@ -733,7 +733,6 @@
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DcbDEBUG" />
 			<Add option="-DCB_PRECOMP" />
-			<Add option="-DWX_PRECOMP" />
 			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#WX.include)" />
 			<Add directory="$(#WX.lib)/gcc_dll$(WX_CFG)/msw$(WX_SUFFIX)" />
Index: src/CodeBlocks_wx30.cbp
===================================================================
--- src/CodeBlocks_wx30.cbp	(revision 9849)
+++ src/CodeBlocks_wx30.cbp	(working copy)
@@ -731,7 +731,7 @@
 			<Add option="-D__WXMSW__" />
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DcbDEBUG" />
-			<Add option="-DNOPCH" />
+			<Add option="-DCB_PRECOMP" />
 			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#WX30.include)" />
 			<Add directory="$(#WX30.lib)/gcc_dll$(WX_CFG)/msw$(WX_SUFFIX)" />
@@ -1283,6 +1283,7 @@
 			<Option target="sdk" />
 		</Unit>
 		<Unit filename="include/sdk.h">
+			<Option compile="1" />
 			<Option weight="1" />
 			<Option target="src" />
 		</Unit>
@@ -1293,6 +1294,7 @@
 			<Option target="sdk" />
 		</Unit>
 		<Unit filename="include/sdk_precomp.h">
+			<Option compile="1" />
 			<Option weight="0" />
 			<Option target="sdk" />
 		</Unit>
Index: src/include/sdk_common.h
===================================================================
--- src/include/sdk_common.h	(revision 9849)
+++ src/include/sdk_common.h	(working copy)
@@ -28,16 +28,28 @@
 
 #if ( defined(CB_PRECOMP) && !defined(WX_PRECOMP) )
     #define WX_PRECOMP
+    // Check for wxWidgets 3.x PCH bug under MinGW GCC 4.8x
+    // If true, undefine WX_PRECOMP to reduce size of PCH file.
+    #ifdef __MINGW32__
+        #if __GNUC__ == 4 && __GNUC_MINOR__ >= 8
+            #include <wx/version.h>
+            #if wxMAJOR_VERSION == 3
+                #undef WX_PRECOMP
+            #endif // wxMAJOR_VERSION
+        #endif // __GNUC__
+    #endif // __MINGW32__
 #endif // CB_PRECOMP
 
 // basic wxWidgets headers : this one itself will check for precompiled headers
 // and if so will include a list of wx headers, at the bottom we add some more headers
 // in the case of precompilation (note : some headers are in both lists)
 // so even if NO CB_PRECOMP we can still have WX_PRECOMP turned on in this "wxprec" header
-#include <wx/wxprec.h>
+#ifdef WX_PRECOMP
+    #include <wx/wxprec.h>
 
-#ifdef __BORLANDC__
-	#pragma hdrstop
+    #ifdef __BORLANDC__
+        #pragma hdrstop
+    #endif
 #endif
 
 #include "prep.h" // this is deliberately not inside the #ifdef block
Index: src/plugins/contrib/wxSmithAui/wxSmithAui.cbp
===================================================================
--- src/plugins/contrib/wxSmithAui/wxSmithAui.cbp	(revision 9849)
+++ src/plugins/contrib/wxSmithAui/wxSmithAui.cbp	(working copy)
@@ -20,6 +20,7 @@
 					<Add option="-ansi" />
 					<Add option="`wx-config --cflags`" />
 					<Add option="-Winvalid-pch" />
+					<Add option="-include sdk.h" />
 					<Add option="-fexceptions" />
 					<Add option="-fPIC" />
 				</Compiler>
@@ -49,6 +50,7 @@
 					<Add option="-fmessage-length=0" />
 					<Add option="-fexceptions" />
 					<Add option="-Winvalid-pch" />
+					<Add option="-include sdk.h" />
 					<Add option="-DHAVE_W32API_H" />
 					<Add option="-D__WXMSW__" />
 					<Add directory="$(#wx.include)" />
@@ -83,7 +85,7 @@
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DcbDEBUG" />
 			<Add option="-DCB_PRECOMP" />
-			<Add option="-DWX_PRECOMP" />
+			<Add directory="../../../.objs/include" />
 			<Add directory="../../../include" />
 			<Add directory="../../../sdk/wxscintilla/include" />
 			<Add directory="../../../sdk/wxpropgrid/include" />
Index: src/plugins/contrib/wxSmithAui/wxSmithAui.cpp
===================================================================
--- src/plugins/contrib/wxSmithAui/wxSmithAui.cpp	(revision 9849)
+++ src/plugins/contrib/wxSmithAui/wxSmithAui.cpp	(working copy)
@@ -1,4 +1,6 @@
+#ifndef SDK_H
 #include <sdk.h> // Code::Blocks SDK
+#endif
 #include <configurationpanel.h>
 #include "wxSmithAui.h"
 
Index: src/plugins/contrib/wxSmithAui/wxSmithAui_wx30.cbp
===================================================================
--- src/plugins/contrib/wxSmithAui/wxSmithAui_wx30.cbp	(revision 9849)
+++ src/plugins/contrib/wxSmithAui/wxSmithAui_wx30.cbp	(working copy)
@@ -21,6 +21,8 @@
 					<Add option="`$(WX_CONFIG) --cflags`" />
 					<Add option="-fexceptions" />
 					<Add option="-fPIC" />
+					<Add option="-Winvalid-pch" />
+					<Add option="-include sdk.h" />
 				</Compiler>
 				<Linker>
 					<Add option="`$(WX_CONFIG) --libs std,aui,propgrid`" />
@@ -46,6 +48,8 @@
 					<Add option="-mthreads" />
 					<Add option="-fmessage-length=0" />
 					<Add option="-fexceptions" />
+					<Add option="-Winvalid-pch" />
+					<Add option="-include sdk.h" />
 					<Add option="-DHAVE_W32API_H" />
 					<Add option="-D__WXMSW__" />
 					<Add directory="$(#WX30.include)" />
@@ -83,7 +87,8 @@
 			<Add option="-DwxUSE_UNICODE" />
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DcbDEBUG" />
-			<Add option="-DNOPCH" />
+			<Add option="-DCB_PRECOMP" />
+			<Add directory="../../../.objs30/include" />
 			<Add directory="../../../include" />
 			<Add directory="../../../sdk/wxscintilla/include" />
 			<Add directory="../wxSmith" />
