Index: src/CodeBlocks.cbp
===================================================================
--- src/CodeBlocks.cbp	(revision 9793)
+++ src/CodeBlocks.cbp	(working copy)
@@ -733,7 +733,6 @@
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DcbDEBUG" />
 			<Add option="-DCB_PRECOMP" />
-			<Add option="-DWX_PRECOMP" />
 			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#WX.include)" />
 			<Add directory="$(#WX.lib)/gcc_dll$(WX_CFG)/msw$(WX_SUFFIX)" />
