Index: src/plugins/contrib/wxSmith/wxSmith-unix.cbp
===================================================================
--- src/plugins/contrib/wxSmith/wxSmith-unix.cbp	(revision 10128)
+++ src/plugins/contrib/wxSmith/wxSmith-unix.cbp	(working copy)
@@ -15,8 +15,12 @@
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --multiple-instance -ns -p debug" />
 				<Option createStaticLib="1" />
+				<Option projectIncludeDirsRelation="2" />
 				<Compiler>
 					<Add option="`wx-config --version=2.8 --cflags`" />
+					<Add option='-include &quot;sdk.h&quot;' />
+					<Add option="-DCB_PRECOMP" />
+					<Add directory="../../../.objs/include" />
 				</Compiler>
 				<Linker>
 					<Add option="`wx-config --version=2.8 --libs`" />
@@ -33,6 +37,7 @@
 				<Option run_host_application_in_terminal="0" />
 				<Compiler>
 					<Add option="`wx-config --version=2.8 --cflags`" />
+					<Add option="-DNOPCH" />
 				</Compiler>
 				<Linker>
 					<Add option="`wx-config --version=2.8 --libs`" />
@@ -54,7 +59,6 @@
 			<Add option="-fPIC" />
 			<Add option='[[if (GetCompilerFactory().GetCompilerVersionString(_T(&quot;gcc&quot;)) &gt;= _T(&quot;4.0.0&quot;)) print(_T(&quot;-Wno-attributes&quot;));]]' />
 			<Add option="-DcbDEBUG" />
-			<Add option="-DCB_PRECOMP" />
 			<Add directory="../../../include" />
 			<Add directory="../../../sdk/wxscintilla/include" />
 			<Add directory="../../../sdk/wxpropgrid/include" />
Index: src/plugins/contrib/wxSmith/wxSmith.cbp
===================================================================
--- src/plugins/contrib/wxSmith/wxSmith.cbp	(revision 10128)
+++ src/plugins/contrib/wxSmith/wxSmith.cbp	(working copy)
@@ -15,6 +15,13 @@
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option createStaticLib="1" />
+				<Option projectIncludeDirsRelation="2" />
+				<Compiler>
+					<Add option='-include &quot;sdk.h&quot;' />
+					<Add option="-DCB_PRECOMP" />
+					<Add option="-DWX_PRECOMP" />
+					<Add directory="../../../.objs/include" />
+				</Compiler>
 			</Target>
 			<Target title="wxSmith">
 				<Option platforms="Windows;" />
@@ -50,8 +57,6 @@
 			<Add option="-D__WXMSW__" />
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DcbDEBUG" />
-			<Add option="-DCB_PRECOMP" />
-			<Add option="-DWX_PRECOMP" />
 			<Add option="-DBUILDING_PLUGIN" />
 			<Add option="-DwxUSE_UNICODE" />
 			<Add directory="$(#WX.lib)\gcc_dll$(WX_CFG)\msw$(WX_SUFFIX)" />
Index: src/plugins/contrib/wxSmith/wxSmith_wx30-unix.cbp
===================================================================
--- src/plugins/contrib/wxSmith/wxSmith_wx30-unix.cbp	(revision 10128)
+++ src/plugins/contrib/wxSmith/wxSmith_wx30-unix.cbp	(working copy)
@@ -15,8 +15,12 @@
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --multiple-instance -ns -p debug" />
 				<Option createStaticLib="1" />
+				<Option projectIncludeDirsRelation="2" />
 				<Compiler>
 					<Add option="`$(WX_CONFIG) --cflags`" />
+					<Add option='-include &quot;sdk.h&quot;' />
+					<Add option="-DCB_PRECOMP" />
+					<Add directory="../../../.objs30/include" />
 				</Compiler>
 				<Linker>
 					<Add option="`$(WX_CONFIG) --libs std,aui,propgrid,richtext`" />
@@ -33,6 +37,7 @@
 				<Option run_host_application_in_terminal="0" />
 				<Compiler>
 					<Add option="`$(WX_CONFIG) --cflags`" />
+					<Add option="-DNOPCH" />
 				</Compiler>
 				<Linker>
 					<Add option="`$(WX_CONFIG) --libs std,propgrid,richtext`" />
@@ -56,7 +61,6 @@
 			<Add option="-fPIC" />
 			<Add option='[[if (GetCompilerFactory().GetCompilerVersionString(_T(&quot;gcc&quot;)) &gt;= _T(&quot;4.0.0&quot;)) print(_T(&quot;-Wno-attributes&quot;));]]' />
 			<Add option="-DcbDEBUG" />
-			<Add option="-DCB_PRECOMP" />
 			<Add directory="../../../include" />
 			<Add directory="../../../sdk/wxscintilla/include" />
 			<Add directory="../../../include/scripting/include" />
