Index: src/plugins/abbreviations/abbreviations.cpp
===================================================================
--- src/plugins/abbreviations/abbreviations.cpp	(revision 9908)
+++ src/plugins/abbreviations/abbreviations.cpp	(working copy)
@@ -11,9 +11,9 @@
 
 #ifndef CB_PRECOMP
     #include <configurationpanel.h>
-    #include <cbstyledtextctrl.h>
-    #include <editorcolourset.h>
 #endif
+#include <cbstyledtextctrl.h>
+#include <editorcolourset.h>
 
 #include "abbreviations.h"
 #include "abbreviationsconfigpanel.h"
Index: src/plugins/abbreviations/abbreviationsconfigpanel.cpp
===================================================================
--- src/plugins/abbreviations/abbreviationsconfigpanel.cpp	(revision 9908)
+++ src/plugins/abbreviations/abbreviationsconfigpanel.cpp	(working copy)
@@ -16,10 +16,10 @@
     #include <wx/choicdlg.h>
 
     #include <configmanager.h>
-    #include <editorcolourset.h>
     #include <editormanager.h>
     #include <logmanager.h>
 #endif
+#include <editorcolourset.h>
 
 #include "cbcolourmanager.h"
 
Index: src/plugins/codecompletion/codecompletion.cpp
===================================================================
--- src/plugins/codecompletion/codecompletion.cpp	(revision 9908)
+++ src/plugins/codecompletion/codecompletion.cpp	(working copy)
@@ -31,7 +31,6 @@
 
     #include <cbeditor.h>
     #include <configmanager.h>
-    #include <editorcolourset.h>
     #include <editormanager.h>
     #include <globals.h>
     #include <logmanager.h>
@@ -40,6 +39,7 @@
     #include <projectmanager.h>
     #include <sdk_events.h>
 #endif
+#include <editorcolourset.h>
 
 #include <wx/tokenzr.h>
 #include <wx/html/htmlwin.h>
Index: src/plugins/codecompletion/coderefactoring.cpp
===================================================================
--- src/plugins/codecompletion/coderefactoring.cpp	(revision 9908)
+++ src/plugins/codecompletion/coderefactoring.cpp	(working copy)
@@ -19,10 +19,10 @@
 
     #include <cbeditor.h>
     #include <cbproject.h>
-    #include <editorcolourset.h>
     #include <editormanager.h>
     #include <logmanager.h>
 #endif
+#include <editorcolourset.h>
 
 #include <wx/progdlg.h>
 
Index: src/plugins/contrib/DoxyBlocks/AutoDoc.cpp
===================================================================
--- src/plugins/contrib/DoxyBlocks/AutoDoc.cpp	(revision 9908)
+++ src/plugins/contrib/DoxyBlocks/AutoDoc.cpp	(working copy)
@@ -26,10 +26,10 @@
   #include <cbeditor.h>
   #include <configmanager.h>
   #include <editormanager.h>
-  #include <editorcolourset.h>
   #include <logmanager.h>
 #endif
 #include <wx/tokenzr.h>
+#include <editorcolourset.h>
 
 #include "DoxyBlocks.h"
 #include "Expressions.h"
Index: src/plugins/contrib/SpellChecker/SpellCheckerPlugin.cpp
===================================================================
--- src/plugins/contrib/SpellChecker/SpellCheckerPlugin.cpp	(revision 9908)
+++ src/plugins/contrib/SpellChecker/SpellCheckerPlugin.cpp	(working copy)
@@ -20,12 +20,12 @@
 #ifndef CB_PRECOMP
     #include <cbeditor.h>
     #include <configmanager.h>
-    #include <editorcolourset.h>
     #include <editormanager.h>
 #endif
 #include <configurationpanel.h>
 #include <editor_hooks.h>
 #include <cbstyledtextctrl.h>
+#include <editorcolourset.h>
 #ifdef wxUSE_STATUSBAR
 #include <cbstatusbar.h>
 #endif
Index: src/plugins/contrib/ThreadSearch/ThreadSearchView.cpp
===================================================================
--- src/plugins/contrib/ThreadSearch/ThreadSearchView.cpp	(revision 9908)
+++ src/plugins/contrib/ThreadSearch/ThreadSearchView.cpp	(working copy)
@@ -24,12 +24,12 @@
 
     #include "cbeditor.h"
     #include "configmanager.h"
-    #include "editorcolourset.h"
     #include "infowindow.h"
     #include "logmanager.h"
 #endif
 
 #include "cbstyledtextctrl.h"
+#include "editorcolourset.h"
 #include "encodingdetector.h"
 #include "SearchInPanel.h"
 #include "DirectoryParamsPanel.h"
Index: src/sdk/cbeditor.cpp
===================================================================
--- src/sdk/cbeditor.cpp	(revision 9908)
+++ src/sdk/cbeditor.cpp	(working copy)
@@ -24,7 +24,6 @@
     #include "cbproject.h"
     #include "configmanager.h"
     #include "debuggermanager.h"
-    #include "editorcolourset.h"
     #include "editormanager.h"
     #include "globals.h"
     #include "infowindow.h"
@@ -38,6 +37,7 @@
     #include "sdk_events.h"
 #endif
 #include "cbstyledtextctrl.h"
+#include "editorcolourset.h"
 #include "cbcolourmanager.h"
 
 #include <wx/fontutil.h>
Index: src/sdk/projectfileoptionsdlg.cpp
===================================================================
--- src/sdk/projectfileoptionsdlg.cpp	(revision 9908)
+++ src/sdk/projectfileoptionsdlg.cpp	(working copy)
@@ -13,7 +13,6 @@
     #include "cbproject.h"
     #include "compilerfactory.h"
     #include "editormanager.h"
-    #include "editorcolourset.h"
     #include "logmanager.h"
     #include "projectmanager.h"
     #include <wx/xrc/xmlres.h>
@@ -29,6 +28,8 @@
     #include <wx/sizer.h>
 #endif
 
+#include "editorcolourset.h"
+
 #ifdef __WXMSW__
 // TODO: equivalent??? -> #include <errno.h>
 #else
Index: src/src/disassemblydlg.cpp
===================================================================
--- src/src/disassemblydlg.cpp	(revision 9908)
+++ src/src/disassemblydlg.cpp	(working copy)
@@ -24,7 +24,6 @@
 
     #include "cbproject.h"
     #include "configmanager.h"
-    #include "editorcolourset.h"
     #include "editormanager.h"
     #include "globals.h"
     #include "manager.h"
@@ -32,6 +31,7 @@
 #endif
 
 #include "debuggermanager.h"
+#include "editorcolourset.h"
 #include "filefilters.h"
 
 // Keep in sync with cbEditor.cpp:
Index: src/src/editkeywordsdlg.cpp
===================================================================
--- src/src/editkeywordsdlg.cpp	(revision 9908)
+++ src/src/editkeywordsdlg.cpp	(working copy)
@@ -16,9 +16,10 @@
     #include <wx/stattext.h>
     #include <wx/textctrl.h>
     #include <wx/wxscintilla.h>
-    #include "editorcolourset.h"
 #endif
 
+#include "editorcolourset.h"
+
 #include "editkeywordsdlg.h"
 
 BEGIN_EVENT_TABLE(EditKeywordsDlg, wxScrollingDialog)
