Index: src/plugins/scriptedwizard/resources/wxwidgets/common/app.cpp
===================================================================
--- src/plugins/scriptedwizard/resources/wxwidgets/common/app.cpp	(revision 9827)
+++ src/plugins/scriptedwizard/resources/wxwidgets/common/app.cpp	(working copy)
@@ -14,7 +14,11 @@
 #ifdef __BORLANDC__
 #pragma hdrstop
 #endif //__BORLANDC__
-
+[IF WXFB]
+#ifndef WX_PRECOMP
+#include <wx/wx.h>
+#endif // WX_PRECOMP
+[ENDIF WXFB]
 #include "[FILENAME_PREFIX]App.h"
 #include "[FILENAME_PREFIX]Main.h"
 
Index: src/plugins/scriptedwizard/resources/wxwidgets/common/main.cpp
===================================================================
--- src/plugins/scriptedwizard/resources/wxwidgets/common/main.cpp	(revision 9827)
+++ src/plugins/scriptedwizard/resources/wxwidgets/common/main.cpp	(working copy)
@@ -14,6 +14,14 @@
 #ifdef __BORLANDC__
 #pragma hdrstop
 #endif //__BORLANDC__
+[IF WXFB]#ifndef WX_PRECOMP
+#include <wx/wx.h>
+#endif // WX_PRECOMP[ENDIF WXFB]
+[IF NONE]#ifndef WX_PRECOMP
+#include <wx/wx.h>
+#endif // WX_PRECOMP[ENDIF NONE]
+[IF WXDIALOG][IF NONE]#include <wx/button.h>
+#include <wx/statline.h>[ENDIF NONE][ENDIF WXDIALOG]
 
 #include "[FILENAME_PREFIX]Main.h"
 
Index: src/plugins/scriptedwizard/resources/wxwidgets/common/main.h
===================================================================
--- src/plugins/scriptedwizard/resources/wxwidgets/common/main.h	(revision 9827)
+++ src/plugins/scriptedwizard/resources/wxwidgets/common/main.h	(working copy)
@@ -10,13 +10,11 @@
 #ifndef [PROJECT_HDR]MAIN_H
 #define [PROJECT_HDR]MAIN_H
 
-[IF NONE]#ifndef WX_PRECOMP
-    #include <wx/wx.h>
-#endif[ENDIF NONE]
+[IF WXFRAME]#include <wx/frame.h>
 
 #include "[FILENAME_PREFIX]App.h"
 
-[IF WXFRAME][IF NONE]class [CLASS_PREFIX]Frame: public wxFrame
+[IF NONE]class [CLASS_PREFIX]Frame: public wxFrame
 {
     public:
         [CLASS_PREFIX]Frame(wxFrame *frame, const wxString& title);
@@ -44,8 +42,14 @@
         virtual void OnQuit(wxCommandEvent& event);
         virtual void OnAbout(wxCommandEvent& event);
 };[ENDIF WXFB][ENDIF WXFRAME]
-[IF WXDIALOG][IF NONE]#include <wx/button.h>
-#include <wx/statline.h>
+[IF WXDIALOG][IF NONE]#include <wx/dialog.h>
+
+#include "[FILENAME_PREFIX]App.h"
+
+class wxButton;
+class wxStaticLine;
+class wxStaticText;
+
 class [CLASS_PREFIX]Dialog: public wxDialog
 {
     public:
