Index: src/sdk/templatemanager.cpp
===================================================================
--- src/sdk/templatemanager.cpp	(revision 9855)
+++ src/sdk/templatemanager.cpp	(working copy)
@@ -263,8 +263,16 @@
         return;
     }
 
+    // get default template title
+    wxString title = prj->GetTitle();
+    
+    // filter title, removing all illegal filename characters
+    wxFileName titleFileName(title) ;
+    wxString forbidden = titleFileName.GetForbiddenChars();
+    for (size_t i=0; i < forbidden.Length(); ++i)
+        title.Replace(wxString(forbidden[i]), wxT(""),true);
+    
     // check if it exists and ask a different title
-    wxString title = prj->GetTitle();
     while (true)
     {
         // ask for template title (unique)
