Index: src/plugins/contrib/SpellChecker/SpellCheckHelper.cpp
===================================================================
--- src/plugins/contrib/SpellChecker/SpellCheckHelper.cpp	(revision 9908)
+++ src/plugins/contrib/SpellChecker/SpellCheckHelper.cpp	(working copy)
@@ -20,12 +20,12 @@
 
 #include <sdk.h> // Code::Blocks SDK
 #ifndef CB_PRECOMP
-    #include <wx/wxscintilla.h>
-
     #include <configmanager.h>
     #include <logmanager.h>
 #endif
 
+#include <wx/wxscintilla.h>
+
 #include "SpellCheckerPlugin.h"
 
 SpellCheckHelper::SpellCheckHelper()
Index: src/plugins/contrib/ToolsPlus/PipedProcessCtrl.h
===================================================================
--- src/plugins/contrib/ToolsPlus/PipedProcessCtrl.h	(revision 9908)
+++ src/plugins/contrib/ToolsPlus/PipedProcessCtrl.h	(working copy)
@@ -11,9 +11,7 @@
 #include <wx/aui/aui.h>
 
 #include <sdk.h>
-#ifndef CB_PRECOMP
-    #include <wx/wxscintilla.h>
-#endif
+#include <wx/wxscintilla.h>
 #include "ShellCtrlBase.h"
 
 class PipedProcessCtrl;
Index: src/sdk/globals.cpp
===================================================================
--- src/sdk/globals.cpp	(revision 9908)
+++ src/sdk/globals.cpp	(working copy)
@@ -18,8 +18,6 @@
     #include <wx/listctrl.h>
     #include <wx/menu.h>
 
-    #include "wx/wxscintilla.h"
-
     #include "cbexception.h"
     #include "configmanager.h" // ReadBool
     #include "filemanager.h"
@@ -28,6 +26,7 @@
     #include "manager.h"
     #include "projectmanager.h"
 #endif
+#include "wx/wxscintilla.h"
 
 #include "tinyxml/tinyxml.h"
 
