Index: src/include/sdk_common.h
===================================================================
--- src/include/sdk_common.h	(revision 9908)
+++ src/include/sdk_common.h	(working copy)
@@ -127,7 +127,6 @@
     #include "compilerfactory.h"
     #include "compiler.h"
     #include "workspaceloader.h"
-    #include "editorcolourset.h"
     #include "pipedprocess.h"
     #include "scrollingdialog.h"
     #include "cbauibook.h"
Index: src/plugins/abbreviations/abbreviations.cpp
===================================================================
--- src/plugins/abbreviations/abbreviations.cpp	(revision 9908)
+++ src/plugins/abbreviations/abbreviations.cpp	(working copy)
@@ -11,9 +11,9 @@
 
 #ifndef CB_PRECOMP
     #include <configurationpanel.h>
-    #include <cbstyledtextctrl.h>
-    #include <editorcolourset.h>
 #endif
+#include <cbstyledtextctrl.h>
+#include <editorcolourset.h>
 
 #include "abbreviations.h"
 #include "abbreviationsconfigpanel.h"
Index: src/plugins/abbreviations/abbreviationsconfigpanel.cpp
===================================================================
--- src/plugins/abbreviations/abbreviationsconfigpanel.cpp	(revision 9908)
+++ src/plugins/abbreviations/abbreviationsconfigpanel.cpp	(working copy)
@@ -16,10 +16,10 @@
     #include <wx/choicdlg.h>
 
     #include <configmanager.h>
-    #include <editorcolourset.h>
     #include <editormanager.h>
     #include <logmanager.h>
 #endif
+#include <editorcolourset.h>
 
 #include "cbcolourmanager.h"
 
Index: src/plugins/codecompletion/codecompletion.cpp
===================================================================
--- src/plugins/codecompletion/codecompletion.cpp	(revision 9908)
+++ src/plugins/codecompletion/codecompletion.cpp	(working copy)
@@ -31,7 +31,6 @@
 
     #include <cbeditor.h>
     #include <configmanager.h>
-    #include <editorcolourset.h>
     #include <editormanager.h>
     #include <globals.h>
     #include <logmanager.h>
@@ -40,6 +39,7 @@
     #include <projectmanager.h>
     #include <sdk_events.h>
 #endif
+#include <editorcolourset.h>
 
 #include <wx/tokenzr.h>
 #include <wx/html/htmlwin.h>
Index: src/plugins/codecompletion/coderefactoring.cpp
===================================================================
--- src/plugins/codecompletion/coderefactoring.cpp	(revision 9908)
+++ src/plugins/codecompletion/coderefactoring.cpp	(working copy)
@@ -19,10 +19,10 @@
 
     #include <cbeditor.h>
     #include <cbproject.h>
-    #include <editorcolourset.h>
     #include <editormanager.h>
     #include <logmanager.h>
 #endif
+#include <editorcolourset.h>
 
 #include <wx/progdlg.h>
 
Index: src/plugins/contrib/envvars/envvars.cpp
===================================================================
--- src/plugins/contrib/envvars/envvars.cpp	(revision 9908)
+++ src/plugins/contrib/envvars/envvars.cpp	(working copy)
@@ -9,21 +9,36 @@
 
 // ----- ----- ----- ----- ----- ----- ----- ----- ----- ----- ----- -----
 
-#include "sdk.h"
+// sdk.h must be the first include in the file to work as a PCH
+#include <sdk.h>
+
+// wxWidgets headers in both wx/wx.h and sdk_common.h
+#if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+    // wxWidgets non GUI headers
+    #include <wx/arrstr.h>
+
+    // wxWidgets GUI headers
+    #include <wx/menu.h>
+    #include <wx/toolbar.h>
+#endif // #if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+
+// CB SDK Headers in sdk_common.h
 #ifndef CB_PRECOMP
-  #include <wx/arrstr.h>
-  #include <wx/menu.h>
-  #include <wx/toolbar.h>
+    // basic headers
+    #include <globals.h>
 
-  #include <tinyxml/tinyxml.h>
+    // absolute base classes
+    #include <cbproject.h>
 
-  #include "cbproject.h"
-  #include "globals.h"
-  #include "manager.h"
-  #include "configmanager.h"
-  #include "logmanager.h"
-#endif
+    // managers
+    #include <manager.h>
+    #include <configmanager.h>
+    #include <logmanager.h>
+#endif // #ifndef CB_PRECOMP
 
+
+#include <tinyxml/tinyxml.h>
+
 #include "projectloader_hooks.h"
 #include <sqplus.h>
 #include <sc_base_types.h>
Index: src/plugins/contrib/envvars/envvars_common.cpp
===================================================================
--- src/plugins/contrib/envvars/envvars_common.cpp	(revision 9908)
+++ src/plugins/contrib/envvars/envvars_common.cpp	(working copy)
@@ -9,19 +9,32 @@
 
 // ----- ----- ----- ----- ----- ----- ----- ----- ----- ----- ----- -----
 
-#include "sdk.h"
-#include "prep.h"
+// sdk.h must be the first include in the file to work as a PCH
+#include <sdk.h>
+
+// wxWidgets headers in both wx/wx.h and sdk_common.h
+#if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+    // wxWidgets non GUI headers
+    #include <wx/utils.h>
+
+    // wxWidgets GUI headers
+    #include <wx/checklst.h>
+#endif // #if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+
+
+// CB SDK Headers in sdk_common.h
 #ifndef CB_PRECOMP
-  #include <wx/checklst.h>
-  #include <wx/utils.h>
+    // basic headers
+    #include <globals.h>
 
-  #include "configmanager.h"
-  #include "globals.h"
-  #include "manager.h"
-  #include "macrosmanager.h"
-  #include "logmanager.h"
-#endif
+    // managers
+    #include <manager.h>
+    #include <configmanager.h>
+    #include <logmanager.h>
+    #include <macrosmanager.h>
+#endif // #ifndef CB_PRECOMP
 
+
 #include <map>
 #include <utility> // std::pair
 
Index: src/plugins/contrib/envvars/envvars_prjoptdlg.cpp
===================================================================
--- src/plugins/contrib/envvars/envvars_prjoptdlg.cpp	(revision 9908)
+++ src/plugins/contrib/envvars/envvars_prjoptdlg.cpp	(working copy)
@@ -9,14 +9,32 @@
 
 // ----- ----- ----- ----- ----- ----- ----- ----- ----- ----- ----- -----
 
-#include "sdk.h"
+// sdk.h must be the first include in the file to work as a PCH
+#include <sdk.h>
+
+// wxWidgets headers in both wx/wx.h and sdk_common.h
+#if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+    // wxWidgets non GUI headers
+
+    // wxWidgets GUI headers
+    #include <wx/checkbox.h>
+    #include <wx/choice.h>
+#endif // #if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+
+// wxWidgets headers in sdk_common.h; but, not in wx/wx.h
+#if !defined(CB_PRECOMP)
+    // wxWidgets non GUI headers
+
+    // wxWidgets GUI headers
+    #include <wx/xrc/xmlres.h>
+#endif // #if !defined(CB_PRECOMP)
+
+// CB SDK Headers in sdk_common.h
 #ifndef CB_PRECOMP
-  #include <wx/xrc/xmlres.h>
-  #include <wx/checkbox.h>
-  #include <wx/choice.h>
+    // absolute base classes
+    #include <cbproject.h>
+#endif // #ifndef CB_PRECOMP
 
-  #include "cbproject.h"
-#endif
 
 #include "envvars.h"
 #include "envvars_common.h"
Index: src/sdk/annoyingdialog.cpp
===================================================================
--- src/sdk/annoyingdialog.cpp	(revision 9908)
+++ src/sdk/annoyingdialog.cpp	(working copy)
@@ -7,20 +7,34 @@
  * $HeadURL$
  */
 
+ // sdk_precomp.h must be the first include in the file to work as a PCH
 #include "sdk_precomp.h"
-#ifndef CB_PRECOMP
+
+// wxWidgets headers in both wx/wx.h and sdk_common.h
+#if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+    // wxWidgets non GUI headers
+    #include <wx/intl.h>
+
+    // wxWidgets GUI headers
     #include <wx/button.h>
     #include <wx/checkbox.h>
-    #include <wx/intl.h>
     #include <wx/sizer.h>
+    #include <wx/statbmp.h>
     #include <wx/stattext.h>
+#endif // #if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+
+// CB SDK Headers in sdk_common.h
+#ifndef CB_PRECOMP
+    // basic headers
+    #include "globals.h"
     #include "cbexception.h"
+
+    // managers
+    #include "manager.h"
     #include "configmanager.h"
-    #include "manager.h"
     #include "logmanager.h"
-    #include "globals.h"
-#endif
-#include <wx/statbmp.h>
+#endif // #ifndef CB_PRECOMP
+
 #include "annoyingdialog.h"
 
 BEGIN_EVENT_TABLE(AnnoyingDialog, wxScrollingDialog)
Index: src/sdk/autodetectcompilers.cpp
===================================================================
--- src/sdk/autodetectcompilers.cpp	(revision 9908)
+++ src/sdk/autodetectcompilers.cpp	(working copy)
@@ -7,27 +7,48 @@
  * $HeadURL$
  */
 
+ // sdk_precomp.h must be the first include in the file to work as a PCH
 #include "sdk_precomp.h"
-#include "autodetectcompilers.h"
 
-#ifndef CB_PRECOMP
+// wxWidgets headers in both wx/wx.h and sdk_common.h
+#if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+    // wxWidgets non GUI headers
+    #include <wx/intl.h>
+    #include <wx/string.h>
+
+    // wxWidgets GUI headers
     #include <wx/button.h>
+    #include <wx/stattext.h>
+#endif // #if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+
+// wxWidgets headers in sdk_common.h; but, not in wx/wx.h
+#if !defined(CB_PRECOMP)
+    // wxWidgets non GUI headers
     #include <wx/filename.h>
-    #include <wx/intl.h>
+
+    // wxWidgets GUI headers
     #include <wx/listctrl.h>
-    #include <wx/stattext.h>
-    #include <wx/string.h>
     #include <wx/xrc/xmlres.h>
+#endif // #if !defined(CB_PRECOMP)
 
+// CB SDK Headers in sdk_common.h
+#ifndef CB_PRECOMP
+    // absolute base classes
+    #include "compilerfactory.h"
     #include "compiler.h"
-    #include "compilerfactory.h"
+
+    // managers
+    #include "manager.h"
     #include "configmanager.h"
-    #include "manager.h"
     #include "macrosmanager.h"
-#endif
+
+    // other base files
+    #include "infowindow.h"
+#endif // #ifndef CB_PRECOMP
+
 #include <wx/tooltip.h>
 
-#include "infowindow.h"
+#include "autodetectcompilers.h"
 
 BEGIN_EVENT_TABLE(AutoDetectCompilers, wxScrollingDialog)
     EVT_UPDATE_UI(-1, AutoDetectCompilers::OnUpdateUI)
Index: src/sdk/cbeditor.cpp
===================================================================
--- src/sdk/cbeditor.cpp	(revision 9908)
+++ src/sdk/cbeditor.cpp	(working copy)
@@ -24,7 +24,6 @@
     #include "cbproject.h"
     #include "configmanager.h"
     #include "debuggermanager.h"
-    #include "editorcolourset.h"
     #include "editormanager.h"
     #include "globals.h"
     #include "infowindow.h"
@@ -38,6 +37,7 @@
     #include "sdk_events.h"
 #endif
 #include "cbstyledtextctrl.h"
+#include "editorcolourset.h"
 #include "cbcolourmanager.h"
 
 #include <wx/fontutil.h>
Index: src/sdk/cbplugin.cpp
===================================================================
--- src/sdk/cbplugin.cpp	(revision 9908)
+++ src/sdk/cbplugin.cpp	(working copy)
@@ -7,20 +7,35 @@
  * $HeadURL$
  */
 
+ // sdk_precomp.h must be the first include in the file to work as a PCH
 #include "sdk_precomp.h"
 
-#ifndef CB_PRECOMP
-    #include <wx/frame.h> // wxFrame
+// wxWidgets headers in both wx/wx.h and sdk_common.h
+#if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+    // wxWidgets non GUI headers
+
+    // wxWidgets GUI headers
+    #include <wx/frame.h>
     #include <wx/menu.h>
+    #include <wx/toolbar.h>
+#endif // #if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+
+// wxWidgets headers in sdk_common.h; but, not in wx/wx.h
+#if !defined(CB_PRECOMP)
+    // wxWidgets non GUI headers
     #include <wx/process.h>
 
+    // wxWidgets GUI headers
+#endif // #if !defined(CB_PRECOMP)
+
+// CB SDK Headers in sdk_common.h
+#ifndef CB_PRECOMP
     #include "cbeditor.h"
     #include "cbplugin.h"
     #include "cbproject.h"
     #include "compiler.h" // GetSwitches
     #include "configmanager.h"
     #include "debuggermanager.h"
-    #include "editorcolourset.h"
     #include "editormanager.h"
     #include "infowindow.h"
     #include "logmanager.h"
@@ -30,8 +45,8 @@
     #include "projectmanager.h"
     #include "sdk_events.h"
 #endif
+#include "editorcolourset.h"
 
-#include <wx/toolbar.h>
 
 #include "annoyingdialog.h"
 #include "cbdebugger_interfaces.h"
Index: src/sdk/compilercommandgenerator.cpp
===================================================================
--- src/sdk/compilercommandgenerator.cpp	(revision 9908)
+++ src/sdk/compilercommandgenerator.cpp	(working copy)
@@ -7,21 +7,45 @@
  * $HeadURL$
  */
 
+ // sdk_precomp.h must be the first include in the file to work as a PCH
 #include "sdk_precomp.h"
+
+// wxWidgets headers in both wx/wx.h and sdk_common.h
+#if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+    // wxWidgets non GUI headers
+    #include <wx/intl.h>
+
+    // wxWidgets GUI headers
+#endif // #if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+
+// wxWidgets headers in sdk_common.h; but, not in wx/wx.h
+#if !defined(CB_PRECOMP)
+    // wxWidgets non GUI headers
+    #include <wx/filename.h>
+
+    // wxWidgets GUI headers
+#endif // #if !defined(CB_PRECOMP)
+
+// CB SDK Headers in sdk_common.h
+#ifndef CB_PRECOMP
+    // basic headers
+    #include "cbexception.h"
+
+    // absolute base classes
+    #include "cbproject.h"
+    #include "compilerfactory.h"
+    #include "compiler.h"
+
+    // managers
+    #include "manager.h"
+    #include "configmanager.h"
+    #include "logmanager.h"
+    #include "scriptingmanager.h"
+    #include "macrosmanager.h"
+#endif // #ifndef CB_PRECOMP
+
 #include "compilercommandgenerator.h"
 
-#include <wx/intl.h>
-#include <wx/filename.h>
-
-#include "cbexception.h"
-#include "cbproject.h"
-#include "compilerfactory.h"
-#include "compiler.h"
-#include "manager.h"
-#include "configmanager.h"
-#include "logmanager.h"
-#include "macrosmanager.h"
-#include "scriptingmanager.h"
 #include "filefilters.h"
 
 #include "scripting/bindings/sc_base_types.h"
Index: src/sdk/globals.cpp
===================================================================
--- src/sdk/globals.cpp	(revision 9908)
+++ src/sdk/globals.cpp	(working copy)
@@ -7,17 +7,32 @@
  * $HeadURL$
  */
 
+ // sdk_precomp.h must be the first include in the file to work as a PCH
 #include "sdk_precomp.h"
 
-#ifndef CB_PRECOMP
+// wxWidgets headers in both wx/wx.h and sdk_common.h
+#if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+    // wxWidgets non GUI headers
+
+    // wxWidgets GUI headers
+    #include <wx/image.h>
+    #include <wx/menu.h>
+    #include <wx/msgdlg.h>
+#endif // #if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+
+// wxWidgets headers in sdk_common.h; but, not in wx/wx.h
+#if !defined(CB_PRECOMP)
+    // wxWidgets non GUI headers
     #include <wx/file.h>
     #include <wx/filename.h>
-    #include <wx/filesys.h>
-    #include <wx/image.h>
+
+    // wxWidgets GUI headers
     #include <wx/imaglist.h>
     #include <wx/listctrl.h>
-    #include <wx/menu.h>
+#endif // #if !defined(CB_PRECOMP)
 
+// CB SDK Headers in sdk_common.h
+#ifndef CB_PRECOMP
     #include "wx/wxscintilla.h"
 
     #include "cbexception.h"
@@ -33,8 +48,8 @@
 
 #include <wx/dirdlg.h>
 #include <wx/filefn.h>
+#include <wx/filesys.h>
 #include <wx/fontmap.h>
-#include <wx/msgdlg.h>
 #include <wx/tokenzr.h>
 
 #include <algorithm>
Index: src/sdk/infowindow.cpp
===================================================================
--- src/sdk/infowindow.cpp	(revision 9908)
+++ src/sdk/infowindow.cpp	(working copy)
@@ -7,22 +7,31 @@
  * $HeadURL$
  */
 
+ // sdk_precomp.h must be the first include in the file to work as a PCH
 #include "sdk_precomp.h"
 
-#ifndef CB_PRECOMP
+// wxWidgets headers in both wx/wx.h and sdk_common.h
+#if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+    // wxWidgets non GUI headers
     #include <wx/intl.h>
-    #include <wx/stattext.h>
+
+    // wxWidgets GUI headers
+    #include <wx/frame.h>
+    #include <wx/panel.h>
     #include <wx/sizer.h>
-    #include <wx/settings.h>
     #include <wx/statbmp.h>
-    #include <wx/panel.h>
-    #include <wx/frame.h>
+    #include <wx/stattext.h>
+#endif // #if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+
+// CB SDK Headers in sdk_common.h
+#ifndef CB_PRECOMP
     #include "infowindow.h"
     #include "manager.h"
 #endif
 
 #include <limits>
 #include <wx/display.h>
+#include <wx/settings.h>
 
 BEGIN_EVENT_TABLE(InfoWindow, wxInfoWindowBase)
 EVT_TIMER(-1, InfoWindow::OnTimer)
Index: src/sdk/manager.cpp
===================================================================
--- src/sdk/manager.cpp	(revision 9908)
+++ src/sdk/manager.cpp	(working copy)
@@ -7,13 +7,30 @@
  * $HeadURL$
  */
 
+ // sdk_precomp.h must be the first include in the file to work as a PCH
 #include "sdk_precomp.h"
 
-#ifndef CB_PRECOMP
-    #include <wx/xrc/xmlres.h>
-    #include <wx/fs_zip.h>
+// wxWidgets headers in both wx/wx.h and sdk_common.h
+#if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+    // wxWidgets non GUI headers
+    #include <wx/app.h> // wxTheApp
+
+    // wxWidgets GUI headers
     #include <wx/menu.h>
+    #include <wx/toolbar.h>
+#endif // #if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
 
+// wxWidgets headers in sdk_common.h; but, not in wx/wx.h
+#if !defined(CB_PRECOMP)
+    // wxWidgets non GUI headers
+    #include <wx/fs_zip.h>
+
+    // wxWidgets GUI headers
+    #include <wx/xrc/xmlres.h>
+#endif // #if !defined(CB_PRECOMP)
+
+// CB SDK Headers in sdk_common.h
+#ifndef CB_PRECOMP
     #include "manager.h" // class's header file
     #include "sdk_events.h"
     #include "cbexception.h"
@@ -33,11 +50,8 @@
     #include "xtra_res.h" // our new ToolBarAddOn handler
 #endif
 
-#include <wx/app.h>    // wxTheApp
-#include <wx/toolbar.h>
 #include <wx/fs_mem.h>
 
-
 #ifdef PPRCESS_EVENT_PERFORMANCE_MEASURE
     // this preprocessor directive can be defined in cbfunctor.h to enable performance measure
     #include <cxxabi.h>  // demangle C++ names
Index: src/sdk/personalitymanager.cpp
===================================================================
--- src/sdk/personalitymanager.cpp	(revision 9908)
+++ src/sdk/personalitymanager.cpp	(working copy)
@@ -7,18 +7,34 @@
  * $HeadURL$
  */
 
+ // sdk_precomp.h must be the first include in the file to work as a PCH
 #include "sdk_precomp.h"
 
+// wxWidgets headers in both wx/wx.h and sdk_common.h
+#if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+    // wxWidgets non GUI headers
+    #include <wx/arrstr.h>
+    #include <wx/intl.h>
+
+    // wxWidgets GUI headers
+#endif // #if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+
+// wxWidgets headers in sdk_common.h; but, not in wx/wx.h
+#if !defined(CB_PRECOMP)
+    // wxWidgets non GUI headers
+    #include <wx/dir.h>
+    #include <wx/filename.h>
+
+    // wxWidgets GUI headers
+#endif // #if !defined(CB_PRECOMP)
+
+// CB SDK Headers in sdk_common.h
 #ifndef CB_PRECOMP
     #include "personalitymanager.h"
     #include "manager.h"
     #include "configmanager.h"
-    #include <wx/intl.h>
-    #include <wx/dir.h>
-    #include <wx/filename.h>
 #endif
 
-#include <wx/arrstr.h>
 
 template<> PersonalityManager* Mgr<PersonalityManager>::instance = nullptr;
 template<> bool  Mgr<PersonalityManager>::isShutdown = false;
Index: src/sdk/pluginmanager.cpp
===================================================================
--- src/sdk/pluginmanager.cpp	(revision 9908)
+++ src/sdk/pluginmanager.cpp	(working copy)
@@ -7,15 +7,33 @@
  * $HeadURL$
  */
 
+ // sdk_precomp.h must be the first include in the file to work as a PCH
 #include "sdk_precomp.h"
 
-#ifndef CB_PRECOMP
-    #include <wx/dir.h>
-    #include <wx/filesys.h>
+// wxWidgets headers in both wx/wx.h and sdk_common.h
+#if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+    // wxWidgets non GUI headers
     #include <wx/intl.h>
-    #include <wx/menu.h>
     #include <wx/string.h>
+    #include <wx/utils.h>
 
+    // wxWidgets GUI headers
+    #include <wx/menu.h>
+#endif // #if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+
+// wxWidgets headers in sdk_common.h; but, not in wx/wx.h
+#if !defined(CB_PRECOMP)
+    // wxWidgets non GUI headers
+    #include <wx/dir.h>
+    #include <wx/filename.h>
+    #include <wx/txtstrm.h>
+    #include <wx/wfstream.h>
+
+    // wxWidgets GUI headers
+#endif // #if !defined(CB_PRECOMP)
+
+// CB SDK Headers in sdk_common.h
+#ifndef CB_PRECOMP
     #include "pluginmanager.h"
     #include "cbexception.h"
     #include "cbplugin.h"
@@ -34,12 +52,8 @@
 #include <wx/dynlib.h>
 #include <wx/filesys.h>
 #include <wx/progdlg.h>
-#include <wx/utils.h>
-#include <wx/filename.h>
 
-#include <wx/wfstream.h>
 #include <wx/zipstrm.h>
-#include <wx/txtstrm.h>
 
 #include "filefilters.h"
 #include "tinyxml/tinyxml.h"
Index: src/sdk/projectfileoptionsdlg.cpp
===================================================================
--- src/sdk/projectfileoptionsdlg.cpp	(revision 9908)
+++ src/sdk/projectfileoptionsdlg.cpp	(working copy)
@@ -13,7 +13,6 @@
     #include "cbproject.h"
     #include "compilerfactory.h"
     #include "editormanager.h"
-    #include "editorcolourset.h"
     #include "logmanager.h"
     #include "projectmanager.h"
     #include <wx/xrc/xmlres.h>
@@ -29,6 +28,8 @@
     #include <wx/sizer.h>
 #endif
 
+#include "editorcolourset.h"
+
 #ifdef __WXMSW__
 // TODO: equivalent??? -> #include <errno.h>
 #else
Index: src/sdk/projectloader.cpp
===================================================================
--- src/sdk/projectloader.cpp	(revision 9908)
+++ src/sdk/projectloader.cpp	(working copy)
@@ -7,15 +7,30 @@
  * $HeadURL$
  */
 
+ // sdk_precomp.h must be the first include in the file to work as a PCH
 #include "sdk_precomp.h"
 
-#ifndef CB_PRECOMP
+// wxWidgets headers in both wx/wx.h and sdk_common.h
+#if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+    // wxWidgets non GUI headers
+    #include <wx/intl.h>
+
+    // wxWidgets GUI headers
+    #include <wx/msgdlg.h>
+#endif // #if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+
+// wxWidgets headers in sdk_common.h; but, not in wx/wx.h
+#if !defined(CB_PRECOMP)
+    // wxWidgets non GUI headers
     #include <wx/confbase.h>
-    #include <wx/fileconf.h>
-    #include <wx/intl.h>
+    #include <wx/dir.h>
     #include <wx/filename.h>
-    #include <wx/msgdlg.h>
-    #include <wx/stopwatch.h>
+
+    // wxWidgets GUI headers
+#endif // #if !defined(CB_PRECOMP)
+
+// CB SDK Headers in sdk_common.h
+#ifndef CB_PRECOMP
     #include "manager.h"
     #include "configmanager.h"
     #include "projectmanager.h"
@@ -26,7 +41,8 @@
     #include "globals.h"
 #endif
 
-#include <wx/dir.h>
+#include <wx/fileconf.h>
+#include <wx/stopwatch.h>
 #include <string>
 
 #include <algorithm>
@@ -34,7 +50,6 @@
 #include "projectloader.h"
 #include "projectloader_hooks.h"
 #include "annoyingdialog.h"
-#include "configmanager.h"
 #include "tinyxml/tinywxuni.h"
 #include "filegroupsandmasks.h"
 
Index: src/sdk/uservarmanager.cpp
===================================================================
--- src/sdk/uservarmanager.cpp	(revision 9908)
+++ src/sdk/uservarmanager.cpp	(working copy)
@@ -7,8 +7,31 @@
  * $HeadURL$
  */
 
+ // sdk_precomp.h must be the first include in the file to work as a PCH
 #include "sdk_precomp.h"
 
+// wxWidgets headers in both wx/wx.h and sdk_common.h
+#if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+    // wxWidgets non GUI headers
+    #include <wx/intl.h>
+
+    // wxWidgets GUI headers
+    #include <wx/button.h>
+    #include <wx/choice.h>
+    #include <wx/textctrl.h>
+    #include <wx/textdlg.h> //wxTextEntryDialog
+#endif // #if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+
+// wxWidgets headers in sdk_common.h; but, not in wx/wx.h
+#if !defined(CB_PRECOMP)
+    // wxWidgets non GUI headers
+
+    // wxWidgets GUI headers
+    #include <wx/splitter.h>
+    #include <wx/xrc/xmlres.h>
+#endif // #if !defined(CB_PRECOMP)
+
+// CB SDK Headers in sdk_common.h
 #ifndef CB_PRECOMP
     #include "uservarmanager.h"
     #include "configmanager.h"
@@ -19,17 +42,10 @@
     #include "cbexception.h"
     #include "infowindow.h"
 
-    #include <wx/button.h>
     #include "scrollingdialog.h"
-    #include <wx/intl.h>
-    #include <wx/xrc/xmlres.h>
-    #include <wx/textctrl.h>
-    #include <wx/splitter.h>
 #endif
 
 #include "annoyingdialog.h"
-#include <wx/choice.h>
-#include <wx/textdlg.h> //wxTextEntryDialog
 
 #if wxCHECK_VERSION(2, 9, 0)
 #include <wx/unichar.h>
Index: src/sdk/workspaceloader.cpp
===================================================================
--- src/sdk/workspaceloader.cpp	(revision 9908)
+++ src/sdk/workspaceloader.cpp	(working copy)
@@ -7,14 +7,28 @@
  * $HeadURL$
  */
 
+ // sdk_precomp.h must be the first include in the file to work as a PCH
 #include "sdk_precomp.h"
 
-#ifndef CB_PRECOMP
-    #include <wx/confbase.h>
-    #include <wx/fileconf.h>
+// wxWidgets headers in both wx/wx.h and sdk_common.h
+#if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+    // wxWidgets non GUI headers
     #include <wx/intl.h>
     #include <wx/string.h>
 
+    // wxWidgets GUI headers
+#endif // #if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+
+// wxWidgets headers in sdk_common.h; but, not in wx/wx.h
+#if !defined(CB_PRECOMP)
+    // wxWidgets non GUI headers
+    #include <wx/confbase.h>
+
+    // wxWidgets GUI headers
+#endif // #if !defined(CB_PRECOMP)
+
+// CB SDK Headers in sdk_common.h
+#ifndef CB_PRECOMP
     #include "workspaceloader.h"
 
     #include "manager.h"
@@ -25,8 +39,8 @@
     #include "cbworkspace.h"
 #endif
 
+#include <wx/fileconf.h>
 
-
 #include "tinyxml/tinyxml.h"
 #include "tinyxml/tinywxuni.h"
 
Index: src/sdk/wxscintilla/include/wx/wxscintilla.h
===================================================================
--- src/sdk/wxscintilla/include/wx/wxscintilla.h	(revision 9908)
+++ src/sdk/wxscintilla/include/wx/wxscintilla.h	(working copy)
@@ -127,6 +127,9 @@
 #define wxSCI_MARKNUM_LASTUNUSED 22
 /* C::B end */
 // Markers used for outlining and changed column.
+/* C::B begin */
+#define wxSCI_MARKNUM_LASTUNUSED 22
+/* C::B end */
 #define wxSCI_MARKNUM_CHANGEUNSAVED 23
 #define wxSCI_MARKNUM_CHANGESAVED 24
 #define wxSCI_MARKNUM_FOLDEREND 25
Index: src/sdk/xtra_res.cpp
===================================================================
--- src/sdk/xtra_res.cpp	(revision 9908)
+++ src/sdk/xtra_res.cpp	(working copy)
@@ -7,14 +7,18 @@
  * $HeadURL$
  */
 
+ // sdk_precomp.h must be the first include in the file to work as a PCH
 #include "sdk_precomp.h"
 
+// CB SDK Headers in sdk_common.h
 #ifndef CB_PRECOMP
-    #include "xtra_res.h"
+    // absolute base classes
     #include "scrollingdialog.h"
-    #include <wx/wx.h>
-#endif
 
+    // other base files
+    #include "xtra_res.h"
+#endif // #ifndef CB_PRECOMP
+
 #include <wx/xml/xml.h>
 
 /////////////////////////////////////////////////////////////////////////////
Index: src/src/app.cpp
===================================================================
--- src/src/app.cpp	(revision 9908)
+++ src/src/app.cpp	(working copy)
@@ -7,44 +7,82 @@
  * $HeadURL$
  */
 
+ // sdk.h must be the first include in the file to work as a PCH
 #include <sdk.h>
-#include "app.h"
-#include <wx/arrstr.h>
-#include <wx/fs_zip.h>
-#include <wx/fs_mem.h>
-#include <wx/xrc/xmlres.h>
+
+// wxWidgets headers in both wx/wx.h and sdk_common.h
+#if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+    // wxWidgets non GUI headers
+    #include <wx/arrstr.h>
+    #include <wx/choicdlg.h>
+    #include <wx/hashmap.h>
+    #include <wx/intl.h>
+    #include <wx/log.h> // for wxSafeShowMessage()
+
+    // wxWidgets GUI headers
+    #include <wx/image.h>
+    #include <wx/msgdlg.h>
+#endif // #if !defined(CB_PRECOMP) && !defined(WX_PRECOMP)
+
+// wxWidgets headers in sdk_common.h; but, not in wx/wx.h
+#if !defined(CB_PRECOMP)
+    // wxWidgets non GUI headers
+    #include <wx/dir.h>
+    #include <wx/filename.h>
+    #include <wx/fs_zip.h>
+    #include <wx/regex.h>
+
+    // wxWidgets GUI headers
+    #include <wx/notebook.h>
+    #include <wx/xrc/xmlres.h>
+#endif // #if !defined(CB_PRECOMP)
+
+// CB SDK Headers in sdk_common.h
+#ifndef CB_PRECOMP
+    // basic headers
+    #include <globals.h>
+    #include <sdk_events.h>
+    #include <cbexception.h>
+
+    // absolute base classes
+    #include <cbauibook.h>
+
+    // managers
+    #include <configmanager.h>
+    #include <editormanager.h>
+    #include <logmanager.h>
+    #include <projectmanager.h>
+    #include <scriptingmanager.h>
+    #include <pluginmanager.h>
+    #include <personalitymanager.h>
+    #include <filemanager.h> // LoaderBase
+
+    // other base files
+    #include <xtra_res.h>
+#endif // #ifndef CB_PRECOMP
+
+// wxWidgets headers in neither wx/wx.h and sdk_common.h
+#include <wx/clipbrd.h>
 #include <wx/cmdline.h>
-#include <wx/regex.h>
+#include <wx/debugrpt.h>
 #include <wx/filefn.h>
-#include <wx/log.h> // for wxSafeShowMessage()
-#include <wx/msgdlg.h>
-#include <wx/choicdlg.h>
-#include <wx/notebook.h>
-#include <wx/clipbrd.h>
+#include <wx/fs_mem.h>
+#include <wx/ipc.h>
 
-#include "cbauibook.h"
-#include <cbexception.h>
-#include <wx/debugrpt.h>
-#include <configmanager.h>
-#include <editormanager.h>
-#include <projectmanager.h>
-#include <personalitymanager.h>
-#include <pluginmanager.h>
-#include <sdk_events.h>
-#include <manager.h>
-#include <scriptingmanager.h>
-#include <globals.h>
-#include <logmanager.h>
+// CB SDK headers not in sdk_common.h
+#include <cbstyledtextctrl.h>
 #include <loggers.h>
-#include "splashscreen.h"
-#include "crashhandler.h"
-#include "cbstyledtextctrl.h"
-#include <wx/ipc.h>
 
-#include "projectmanagerui.h"
-
+//
 #include <sqplus.h>
 
+// Local header for this source file
+#include "app.h"
+
+// Other local header
+#include "projectmanagerui.h"
+#include "splashscreen.h"
+#include "crashhandler.h"
 #ifndef __WXMSW__
     #include "prefix.h" // binreloc
 #endif
@@ -57,11 +95,6 @@
 
 #include "appglobals.h"
 
-#ifndef CB_PRECOMP
-    #include <wx/dir.h>
-    #include "xtra_res.h"
-    #include "filemanager.h" // LoaderBase
-#endif
 
 #ifndef APP_PREFIX
 #define APP_PREFIX ""
Index: src/src/disassemblydlg.cpp
===================================================================
--- src/src/disassemblydlg.cpp	(revision 9908)
+++ src/src/disassemblydlg.cpp	(working copy)
@@ -24,7 +24,6 @@
 
     #include "cbproject.h"
     #include "configmanager.h"
-    #include "editorcolourset.h"
     #include "editormanager.h"
     #include "globals.h"
     #include "manager.h"
@@ -32,6 +31,7 @@
 #endif
 
 #include "debuggermanager.h"
+#include "editorcolourset.h"
 #include "filefilters.h"
 
 // Keep in sync with cbEditor.cpp:
Index: src/src/editkeywordsdlg.cpp
===================================================================
--- src/src/editkeywordsdlg.cpp	(revision 9908)
+++ src/src/editkeywordsdlg.cpp	(working copy)
@@ -16,9 +16,10 @@
     #include <wx/stattext.h>
     #include <wx/textctrl.h>
     #include <wx/wxscintilla.h>
-    #include "editorcolourset.h"
 #endif
 
+#include "editorcolourset.h"
+
 #include "editkeywordsdlg.h"
 
 BEGIN_EVENT_TABLE(EditKeywordsDlg, wxScrollingDialog)
