diff --git include/wx/fontutil.h include/wx/fontutil.h
index 1790a5b..678beef 100644
--- include/wx/fontutil.h
+++ include/wx/fontutil.h
@@ -25,6 +25,10 @@
     #include "wx/msw/wrapwin.h"
 #endif
 
+#if defined(__WXQT__)
+    class QFont;
+#endif
+
 class WXDLLIMPEXP_FWD_BASE wxArrayString;
 struct WXDLLIMPEXP_FWD_CORE wxNativeEncodingInfo;
 
diff --git include/wx/qt/private/converter.h include/wx/qt/private/converter.h
index 5bc0f2d..208ac86 100644
--- include/wx/qt/private/converter.h
+++ include/wx/qt/private/converter.h
@@ -14,6 +14,8 @@
 #include "wx/defs.h"
 #include <QtCore/Qt>
 
+#include "wx/kbdstate.h"
+
 // Rely on overloading and let the compiler pick the correct version, which makes
 // them easier to use then to write wxQtConvertQtRectToWxRect() or wxQtConvertWxRectToQtRect()
 
diff --git include/wx/qt/private/winevent.h include/wx/qt/private/winevent.h
index 45e94f0..286a6b0 100644
--- include/wx/qt/private/winevent.h
+++ include/wx/qt/private/winevent.h
@@ -11,6 +11,7 @@
 #ifndef _WX_QT_EVENTSIGNALFORWARDER_H_
 #define _WX_QT_EVENTSIGNALFORWARDER_H_
 
+#include "wx/log.h"
 #include "wx/window.h"
 #include "wx/qt/private/converter.h"
 #include "wx/qt/private/utils.h"
diff --git include/wx/qt/toolbar.h include/wx/qt/toolbar.h
index 276f7ff..63b76f8 100644
--- include/wx/qt/toolbar.h
+++ include/wx/qt/toolbar.h
@@ -10,6 +10,7 @@
 #ifndef _WX_QT_TOOLBAR_H_
 #define _WX_QT_TOOLBAR_H_
 
+class QActionGroup;
 class wxQtToolBar;
 
 class WXDLLIMPEXP_CORE wxToolBar : public wxToolBarBase
diff --git src/common/fontcmn.cpp src/common/fontcmn.cpp
index 4f0a4a6..1b5f8db 100644
--- src/common/fontcmn.cpp
+++ src/common/fontcmn.cpp
@@ -39,6 +39,10 @@
     #include  "wx/msw/winundef.h"
 #endif
 
+#if defined(__WXQT__)
+    #include <QtGui/QFont> // needed before wx/fontutil.h
+#endif
+
 #include "wx/fontutil.h" // for wxNativeFontInfo
 #include "wx/fontmap.h"
 #include "wx/fontenum.h"
diff --git src/common/fontmap.cpp src/common/fontmap.cpp
index ffac05c..bb8aca0 100644
--- src/common/fontmap.cpp
+++ src/common/fontmap.cpp
@@ -25,6 +25,10 @@
 
 #if wxUSE_FONTMAP
 
+#if defined(__WXQT__)
+    #include <QtGui/QFont> // needed before wx/fontmap.h
+#endif
+
 #include "wx/fontmap.h"
 
 #ifndef WX_PRECOMP
diff --git src/common/fontutilcmn.cpp src/common/fontutilcmn.cpp
index f13e717..aa6f05f 100644
--- src/common/fontutilcmn.cpp
+++ src/common/fontutilcmn.cpp
@@ -23,6 +23,10 @@
     #pragma hdrstop
 #endif
 
+#if defined(__WXQT__)
+    #include <QtGui/QFont> // needed before wx/fontutil.h
+#endif
+
 #include "wx/fontutil.h"
 #include "wx/encinfo.h"
 
diff --git src/html/helpdata.cpp src/html/helpdata.cpp
index 35d0522..91b003e 100644
--- src/html/helpdata.cpp
+++ src/html/helpdata.cpp
@@ -24,6 +24,10 @@
 
 #include <ctype.h>
 
+#if defined(__WXQT__)
+    #include <QtGui/QFont> // needed before wx/fontmap.h
+#endif
+
 #include "wx/html/helpdata.h"
 #include "wx/tokenzr.h"
 #include "wx/wfstream.h"
diff --git src/html/htmlpars.cpp src/html/htmlpars.cpp
index d0663f5..e48aa9a 100644
--- src/html/htmlpars.cpp
+++ src/html/htmlpars.cpp
@@ -22,6 +22,10 @@
     #include "wx/wxcrtvararg.h"
 #endif
 
+#if defined(__WXQT__)
+    #include <QtGui/QFont> // needed before wx/fontmap.h
+#endif
+
 #include "wx/tokenzr.h"
 #include "wx/wfstream.h"
 #include "wx/url.h"
diff --git src/qt/bitmap.cpp src/qt/bitmap.cpp
index 2265de7..97e9fd0 100644
--- src/qt/bitmap.cpp
+++ src/qt/bitmap.cpp
@@ -8,15 +8,24 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+#include <QtGui/QPixmap>
+#include <QtGui/QBitmap>
+#include <QtWidgets/QLabel>
+
+#ifndef WX_PRECOMP
+    #include "wx/icon.h"
+    #include "wx/image.h"
+#endif // WX_PRECOMP
+
 #include "wx/bitmap.h"
-#include "wx/image.h"
 #include "wx/rawbmp.h"
 #include "wx/qt/private/converter.h"
 #include "wx/qt/private/utils.h"
 
-#include <QtGui/QPixmap>
-#include <QtGui/QBitmap>
-#include <QtWidgets/QLabel>
 
 static wxImage ConvertImage( QImage qtImage )
 {
diff --git src/qt/button.cpp src/qt/button.cpp
index d06e145..382f126 100644
--- src/qt/button.cpp
+++ src/qt/button.cpp
@@ -8,8 +8,16 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+#ifndef WX_PRECOMP
+    #include "wx/bitmap.h"
+#endif // WX_PRECOMP
+
 #include "wx/button.h"
-#include "wx/bitmap.h"
+#include "wx/stockitem.h"
 #include "wx/qt/private/utils.h"
 #include "wx/qt/private/winevent.h"
 
diff --git src/qt/clipbrd.cpp src/qt/clipbrd.cpp
index 438ee99..3a5ec3b 100644
--- src/qt/clipbrd.cpp
+++ src/qt/clipbrd.cpp
@@ -8,12 +8,18 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+#include <QApplication>
+#include <QClipboard>
+
 #include "wx/clipbrd.h"
 #include "wx/scopedarray.h"
 #include "wx/scopeguard.h"
 #include "wx/qt/private/converter.h"
 
-#include <QClipboard>
 
 // ----------------------------------------------------------------------------
 // wxClipboard ctor/dtor
diff --git src/qt/colour.cpp src/qt/colour.cpp
index e64784e..bad77af 100644
--- src/qt/colour.cpp
+++ src/qt/colour.cpp
@@ -9,12 +9,18 @@
 
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
-#include "wx/qt/private/utils.h"
 
 #ifdef __BORLANDC__
-#pragma hdrstop
+    #pragma hdrstop
 #endif
 
+#ifndef WX_PRECOMP
+    #include "wx/colour.h"
+#endif // WX_PRECOMP
+
+#include "wx/qt/private/utils.h"
+
+
 int wxColour::GetPixel() const
 {
     wxMISSING_IMPLEMENTATION( "wxColour::GetPixel" );
diff --git src/qt/converter.cpp src/qt/converter.cpp
index 5b407f5..334c41d 100644
--- src/qt/converter.cpp
+++ src/qt/converter.cpp
@@ -9,8 +9,9 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
-#include "wx/gdicmn.h"
-#include "wx/gdicmn.h"
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
 
 #include <QtCore/QRect>
 #include <QtCore/QString>
@@ -22,6 +23,11 @@
     #include <QtCore/QDate>
 #endif // wxUSE_DATETIME
 
+#include "wx/kbdstate.h"
+#include "wx/gdicmn.h"
+#include "wx/gdicmn.h"
+
+
 wxPoint wxQtConvertPoint( const QPoint &point )
 {
     if (point.isNull())
diff --git src/qt/cursor.cpp src/qt/cursor.cpp
index edefb18..0fb633c 100644
--- src/qt/cursor.cpp
+++ src/qt/cursor.cpp
@@ -7,11 +7,21 @@
 
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
-#include "wx/cursor.h"
+
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
 
 #include <QApplication>
 #include <QBitmap>
 
+#ifndef WX_PRECOMP
+    #include "wx/bitmap.h"
+#endif // WX_PRECOMP
+
+#include "wx/cursor.h"
+
+
 void wxSetCursor(const wxCursor& cursor)
 {
     if (cursor.m_qtCursor.shape() == Qt::ArrowCursor)
diff --git src/qt/dataobj.cpp src/qt/dataobj.cpp
index b406310..0b47289 100644
--- src/qt/dataobj.cpp
+++ src/qt/dataobj.cpp
@@ -8,6 +8,12 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+#include <QObject>
+
 #include "wx/qt/private/converter.h"
 #include "wx/qt/private/utils.h"
 #include "wx/dataobj.h"
diff --git src/qt/dc.cpp src/qt/dc.cpp
index 807eff0..14cde6f 100644
--- src/qt/dc.cpp
+++ src/qt/dc.cpp
@@ -8,12 +8,22 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+#include <QtGui/QBitmap>
+
+#ifndef WX_PRECOMP
+    #include "wx/icon.h"
+    #include "wx/log.h"
+#endif // WX_PRECOMP
+
 #include "wx/dc.h"
-#include "wx/icon.h"
 #include "wx/qt/dc.h"
 #include "wx/qt/private/converter.h"
 #include "wx/qt/private/utils.h"
-#include <QtGui/QBitmap>
+
 
 static void SetPenColour( QPainter *qtPainter, QColor col )
 {
diff --git src/qt/dcclient.cpp src/qt/dcclient.cpp
index b7e7270..3b8dfce 100644
--- src/qt/dcclient.cpp
+++ src/qt/dcclient.cpp
@@ -8,11 +8,20 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+#include <QtGui/QPicture>
+
+#ifndef WX_PRECOMP
+    #include "wx/log.h"
+    #include "wx/window.h"
+#endif // WX_PRECOMP
+
 #include "wx/dcclient.h"
-#include "wx/log.h"
 #include "wx/qt/dcclient.h"
 
-#include <QtGui/QPicture>
 
 //##############################################################################
 
diff --git src/qt/font.cpp src/qt/font.cpp
index 41e7673..f9d5525 100644
--- src/qt/font.cpp
+++ src/qt/font.cpp
@@ -8,13 +8,18 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+#include <QtGui/QFont>
+#include <QtGui/QFontInfo>
+
 #include "wx/font.h"
 #include "wx/fontutil.h"
 #include "wx/qt/private/utils.h"
 #include "wx/qt/private/converter.h"
 
-#include <QtGui/QFont>
-#include <QtGui/QFontInfo>
 
 static QFont::StyleHint ConvertFontFamily(wxFontFamily family)
 {
diff --git src/qt/fontutil.cpp src/qt/fontutil.cpp
index d77d7dc..3ce79b0 100644
--- src/qt/fontutil.cpp
+++ src/qt/fontutil.cpp
@@ -8,6 +8,12 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+#include <QtGui/QFont>
+
 #include "wx/fontutil.h"
 #include "wx/encinfo.h"
 
diff --git src/qt/frame.cpp src/qt/frame.cpp
index 4944181..5f39c9f 100644
--- src/qt/frame.cpp
+++ src/qt/frame.cpp
@@ -8,12 +8,21 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+#include <QtWidgets/QScrollArea>
+
+#ifndef WX_PRECOMP
+    #include "wx/menu.h"
+    #include "wx/toolbar.h"
+#endif // WX_PRECOMP
+
 #include "wx/frame.h"
-#include "wx/menu.h"
 #include "wx/qt/private/utils.h"
 #include "wx/qt/private/converter.h"
 #include "wx/qt/private/winevent.h"
-#include <QtWidgets/QScrollArea>
 
 
 class wxQtMainWindow : public wxQtEventSignalHandler< QMainWindow, wxFrame >
diff --git src/qt/listctrl.cpp src/qt/listctrl.cpp
index 9c3ccbc..3fced62 100644
--- src/qt/listctrl.cpp
+++ src/qt/listctrl.cpp
@@ -8,12 +8,20 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+#include <QHeaderView>
+
+#ifndef WX_PRECOMP
+    #include "wx/bitmap.h"
+#endif // WX_PRECOMP
+
 #include "wx/listctrl.h"
 #include "wx/imaglist.h"
 #include "wx/qt/private/winevent.h"
 
-#include <QHeaderView>
-
 
 class wxQtTreeWidget : public wxQtEventSignalHandler< QTreeWidget, wxListCtrl >
 {
diff --git src/qt/tglbtn.cpp src/qt/tglbtn.cpp
index d4cbc11..a83b247 100644
--- src/qt/tglbtn.cpp
+++ src/qt/tglbtn.cpp
@@ -8,8 +8,16 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+#ifndef WX_PRECOMP
+    #include "wx/bitmap.h"
+#endif // WX_PRECOMP
+
+#include "wx/stockitem.h"
 #include "wx/tglbtn.h"
-#include "wx/bitmap.h"
 #include "wx/qt/private/converter.h"
 #include "wx/qt/private/winevent.h"
 
diff --git src/qt/toolbar.cpp src/qt/toolbar.cpp
index ed8beca..6240986 100644
--- src/qt/toolbar.cpp
+++ src/qt/toolbar.cpp
@@ -8,14 +8,23 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
-#include "wx/toolbar.h"
-#include "wx/qt/private/winevent.h"
-#include "wx/qt/private/converter.h"
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
 
 #if wxUSE_TOOLBAR
 
+#include <QActionGroup>
 #include <QtWidgets/QToolButton>
 
+#ifndef WX_PRECOMP
+    #include "wx/menu.h"
+#endif // WX_PRECOMP
+
+#include "wx/toolbar.h"
+#include "wx/qt/private/winevent.h"
+#include "wx/qt/private/converter.h"
+
 class wxQtToolButton;
 class wxToolBarTool : public wxToolBarToolBase
 {
diff --git src/qt/tooltip.cpp src/qt/tooltip.cpp
index 409035a..f78b44f 100644
--- src/qt/tooltip.cpp
+++ src/qt/tooltip.cpp
@@ -8,6 +8,14 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+#ifndef WX_PRECOMP
+    #include "wx/window.h"
+#endif // WX_PRECOMP
+
 #include "wx/tooltip.h"
 #include "wx/qt/private/utils.h"
 
diff --git src/qt/uiaction.cpp src/qt/uiaction.cpp
index b37d946..1278434 100644
--- src/qt/uiaction.cpp
+++ src/qt/uiaction.cpp
@@ -8,12 +8,17 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
+
+#include <QtTest/QtTestGui>
+
 #include "wx/qt/defs.h"
 #include "wx/qt/private/utils.h"
 #include "wx/uiaction.h"
 #include "wx/qt/private/converter.h"
 
-#include <QtTest/QtTestGui>
 
 #if wxUSE_UIACTIONSIMULATOR
 
diff --git src/qt/utils.cpp src/qt/utils.cpp
index cca032f..786264b 100644
--- src/qt/utils.cpp
+++ src/qt/utils.cpp
@@ -8,11 +8,9 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
-#include "wx/utils.h"
-#include "wx/window.h"
-#include "wx/cursor.h"
-#include "wx/qt/private/utils.h"
-#include "wx/qt/private/converter.h"
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
 
 #include <QtGui/QCursor>
 #include <QtWidgets/QApplication>
@@ -20,6 +18,17 @@
 #include <QtGui/QDesktopServices>
 #include <QtCore/QUrl>
 
+#ifndef WX_PRECOMP
+    #include "wx/cursor.h"
+    #include "wx/log.h"
+    #include "wx/window.h"
+#endif // WX_PRECOMP
+
+#include "wx/utils.h"
+#include "wx/qt/private/utils.h"
+#include "wx/qt/private/converter.h"
+
+
 void wxMissingImplementation( const char fileName[], unsigned lineNumber,
     const char feature[] )
 {
diff --git src/qt/window.cpp src/qt/window.cpp
index 31969dc..c267af2 100644
--- src/qt/window.cpp
+++ src/qt/window.cpp
@@ -8,14 +8,9 @@
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
-#include "wx/window.h"
-#include "wx/log.h"
-#include "wx/menu.h"
-#include "wx/tooltip.h"
-#include "wx/scrolbar.h"
-#include "wx/qt/private/utils.h"
-#include "wx/qt/private/converter.h"
-#include "wx/qt/private/winevent.h"
+#ifdef __BORLANDC__
+    #pragma hdrstop
+#endif
 
 #include <QtGui/QPicture>
 #include <QtGui/QPainter>
@@ -24,6 +19,20 @@
 #include <QtWidgets/QWidget>
 #include <QtWidgets/QScrollArea>
 
+#ifndef WX_PRECOMP
+    #include "wx/dcclient.h"
+    #include "wx/frame.h"
+    #include "wx/log.h"
+    #include "wx/menu.h"
+    #include "wx/scrolbar.h"
+#endif // WX_PRECOMP
+
+#include "wx/window.h"
+#include "wx/tooltip.h"
+#include "wx/qt/private/utils.h"
+#include "wx/qt/private/converter.h"
+#include "wx/qt/private/winevent.h"
+
 
 #define VERT_SCROLLBAR_POSITION 0, 1
 #define HORZ_SCROLLBAR_POSITION 1, 0
